---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%",
  warning = FALSE,
  message = FALSE
)
library(gnomeR)
library(knitr)
library(dplyr)
```

# gnomeR

<!-- badges: start -->
[![Travis build status](https://travis-ci.com/AxelitoMartin/gnomeR.svg?branch=development)](https://travis-ci.org/AxelitoMartin/gnomeR)
[![Codecov test coverage](https://codecov.io/gh/AxelitoMartin/gnomeR/branch/development/graph/badge.svg)](https://codecov.io/gh/AxelitoMartin/gnomeR?branch=development)
<!-- badges: end -->

the `gnomeR` package provides a consistent framework for genetic data processing, visualization and analysis. This is primarily targeted to IMPACT datasets but can also be applied to any genomic data provided by CbioPortal. 

  - [**Dowloading and gathering data from CbioPortal**](https://axelitomartin.github.io/gnomeR/articles/API-tutorial.html) through an integrated API using simply the sample IDs of the samples of interests or the name of the study to retrive all samples in that study. 
  - [**Processing genomic data**](https://axelitomartin.github.io/gnomeR/articles/Data-processing.html) retrieved for mutations (MAF file), fusions (MAF file) and copy-number alterations (and when available segmentation files) into an analysis ready format. 
  - [**Visualization of the processed data**](https://axelitomartin.github.io/gnomeR/articles/Visualizations.html) provided through MAF file summaries, OncoPrints and heatmaps.
  - [**Analyzing the processed data**](https://axelitomartin.github.io/gnomeR/articles/Analizing-genomic-data.html) for association with binary, continuous and survival outcome. Including further visualiztion to improve understanding of the results. 
  
## Installation

You can install the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("AxelitoMartin/gnomeR")
```

## Examples

### Setting up the API

In order to download the data from CbioPortal, one must first require a token from the website [CbioPortal](https://cbioportal.mskcc.org/) wich will prompt a login page with your MSKCC credentials. Then navigate to "Web API" in the top bar menu, following this simply download a token and copy it after running the following command in R:


```{r,eval=F}
usethis::edit_r_environ()
```

And pasting the token you were given in the .Renviron file that was created and saving after pasting your token.

```{r, eval=F}
CBIOPORTAL_TOKEN = 'YOUR_TOKEN'
```

You can test your connection using:

```{r,eval = F}
get_cbioportal_token()
```


### Retrieving data

Now that the Cbioportal API is set up in your environment, you must first specify the database of interest (IMPACT or TCGA are the two available options). Following this one can either sepcify the samples or study of interest:

```{r, eval = F}
ids <- as.character(unique(mut$Tumor_Sample_Barcode)[1:100])
df <- get_genetics(sample_ids = ids,database = "msk_impact",
                       mutations = TRUE, fusions = TRUE, cna = TRUE)
```


### Processing the downloaded data 

The `binmat()` function is the feature of the data processing of `gnomeR`. It takes genomic inputs from various sources of CbioPortal (mutation files, fusion files and copy number raw counts) to give out a clean binary matrix of n samples by all the events that were found in the files.


```{r, eval = F}
df.clean <- binmat(maf = df$mut, cna = df$cna)
```

We further included example datasets from the raw dowloaded files on CbioPortal (`mut`, `fusion`, `cna`) which we will use for the following examples.

```{r}
set.seed(123)
patients <- as.character(unique(mut$Tumor_Sample_Barcode))[sample(1:length(unique(mut$Tumor_Sample_Barcode)), 100, replace=FALSE)]

gen.dat <- binmat(patients = patients, maf = mut, fusion = fusion, cna = cna)
kable(gen.dat[1:10,1:10],row.names = T)
```

### Visualization 

#### MAF 

Before we move on to more complex visualizations, we integrate the `maf.summary()` function to give an overview of the distribution of the different mutations across the cohort of interest:

```{r}
sum.plots <- maf.summary(maf = mut %>% filter(Tumor_Sample_Barcode %in% patients))
sum.plots$p.genes
sum.plots$p.comut
```


#### OncoPrints

OncoPrints are a convenient way to display the overall genomic profiles of samples in the cohort of interest. This is best used for a subset of genes that are under consideration.

```{r}
genes <- c("TP53","PIK3CA","KRAS","TERT","EGFR","FAT","ALK","CDKN2A","CDKN2B")
plot_oncoPrint(gen.dat = gen.dat %>% select(starts_with(genes)))
```

#### FACETs 

[FACETs](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/) is an ASCN tool and open-source software with a broad application to whole genome, whole-exome, as well as targeted panel sequencing platforms. It is a fully integrated stand-alone pipeline that includes sequencing BAM file post-processing, joint segmentation of total- and allele-specific read counts, and integer copy number calls corrected for tumor purity, ploidy and clonal heterogeneity, with comprehensive output.

```{r}
# need to find how to get facets from API #
# also makes me think we need to get the clinical data too in that for purity #
p.heat <- facets.heatmap(seg = seg, patients = patients, min.purity = 0)
p.heat$p
# Also need a better example lol#
```


### Analysis

In this section we will quickly overview the possible analysis in gnomeR.

#### Binary and continuous outcomes

The `gen.tab()` function let's the user perform a large scale association between the genomic features present in the `binat()` function output and an outcome of choice:

- binary (unpaired test using Fisher's exact test and paired test using McNemmar's exact test)
- continuous (using simple linear regression)

```{r}
outcome <- factor(rbinom(n = length(patients),size = 1,prob = 1/2),levels = c("0","1"))
out <- gen.tab(gen.dat = gen.dat,outcome = outcome,filter = 0.05)
kable(out$fits[1:10,],row.names = T)
out$forest.plot
out$vPlot
```

#### Survival analysis 

##### Univariate 

```{r}
time <- rexp(length(patients))
status <- outcome
surv.dat <- as.data.frame(cbind(time,status))
out <- uni.cox(X = gen.dat, surv.dat = surv.dat, surv.formula = Surv(time,status)~.,filter = 0.05)
kable(out$tab[1:10,],row.names = T)
# out$p
out$KM[[1]]
```
