---
title: "Data-processing"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Data-processing}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>", warning = F, message = F
)
```

```{r, message = F, warning = F}
library(gnomeR)
```

```{r, echo=F}
library(tibble)
library(knitr)
library(dplyr)
set.seed(123)
```

# Introduction 

The main purpose of `gnomeR` is to streamline the processing of genomic files provided by CbioPortal. If you wish to learn how to use the integrated API please read the `API-tutorial` article first. The core function of the processing of these files is performed by the `binmat()` function. It takes the following arguments:

- patients: A character vector that let's the user specify the patients to be used to create the matrix. Default is NULL, in which case all samples found in the provided genetic files will be used.
- maf: A MAF file.
- mut.type: The mutation type to be used. Options are "SOMATIC", "GERMLINE" or "ALL". Note "ALL" will keep all mutations regardless of status (not recommended). Default is SOMATIC.
- SNP.only Boolean to rather the genetics events to be kept only to be SNPs (insertions and deletions will be removed). Default is FALSE.
- include.silent: Boolean to keep or remove all silent mutations. TRUE keeps, FALSE removes. Default is FALSE.
- fusion: An optional MAF file for fusions. If inputed the outcome will be added to the matrix with columns ending in ".fus". Default is NULL. Note if fusions are found in the MAF file in the previous input then these will be added automatically.
- cna: An optional CNA files. If inputed the outcome will be added to the matrix with columns ending in ".cna", .del" and ".amp" (depending on subsequent arguments).
- cna.binary: A boolean argument specifying if the cna events should be enforced as binary. In which case separate columns for amplifications and deletions will be created. Default is FALSE in which case columns ending in ".cna" will be added to the output.
- cna.relax: If cna.binary is TRUE, cna data only enables to count both gains and shallow deletions as amplifications and deletions respectively.
- specify.plat: boolean specifying if specific IMPACT platforms should be considered. When TRUE NAs will fill the cells for genes of patients that were not sequenced on that plaform. Default is TRUE.
- set.plat character argument specifying which IMPACT platform the data should be reduced to if specify.plat is set to TRUE. Options are "341" and "410". Default is NULL.
- rm.empty: boolean specifying if columns with no events founds should be removed. Default is TRUE.
- pathway boolean specifying if pathway annotation should be applied. If TRUE, the function will return a supplementary binary dataframe with columns being each pathway and each row being a sample. Default is FALSE.
- col.names character vector of the necessary columns to be used. By default: col.names = c(Tumor_Sample_Barcode = NULL, Hugo_Symbol = NULL, Variant_Classification = NULL, Mutation_Status = NULL, Variant_Type = NULL)
- oncokb boolean specfiying if maf file should be oncokb annotated. Variants found to be 'Oncogenic' and 'Likely Oncogenic' will be kept. Default is FALSE.

This function returns a matrix containing all the genetic information with rows as samples and columns as features.
A warning will be thrown if some samples were found to have no mutations in the MAF file.

In the follwing sections we will present examples to process each of the datatypes in cbioportal.

# Processing genetic data

## Mutations

The most commmon type of genetic features used in genomic studies at MSKCC. The IMPACT sequencing panel consist of a curated list of genes that are known to have cancer related properties when altered. You can find a complete list of these genes and which platform they were added on in the `impact_genes` datafile.
Note that the MAF files provided through the API embedded in this package and the one provided through raw download on the website are slightly different and we will thus present examples using both these datasets.

### Processing raw mutation data

We included in `gnomeR` an example of raw downloaded MAF file directly from the website in the `mut` dataset. We show here an example selecting a random subset of 100 samples in the `mut` dataset:

```{r}
as_tibble(mut)
samples <- as.character(unique(mut$Tumor_Sample_Barcode))[sample(1:length(unique(mut$Tumor_Sample_Barcode)), 100, replace=FALSE)]
df <- binmat(patients = samples ,maf = mut)
kable(df[1:10, 1:10])
```

Note that by default in the situation above the outputted dataframe is a binary matrix made from all types of mutations and adjusting the features for the platform they were added on. Thus all samples that were sequenced on the original platform have NA's in the cells of for features that were added on subsequent platforms.
In the case where the user plans on using methods that do not accept missing values, the `specify.plat` argument can be changed to FALSE to replace all the NA's mentioned above to 0. We show below such an example, we moreover make this example including only SNPs (including silent mutations):

```{r}
df <- binmat(patients = samples ,maf = mut, SNP.only = TRUE, include.silent = TRUE, specify.plat = FALSE)
kable(df[1:10, 1:10])
```

### Processing mutation data from the API

In this section we show the same examples as above but using the data downloaded through the API:

```{r, results="hide"}
# samples.tcga <- tcga_samples$patient_id[1:50]
# dat <- get_genetics(sample_ids = samples.tcga, database = 'tcga', fusions = FALSE, cna = FALSE)
# df <- binmat(maf = dat$mut)
```

```{r}
# kable(df[1:10, 1:10])
```

Similarly,

```{r}
# df <- binmat(maf = dat$mut, SNP.only = TRUE, include.silent = TRUE, specify.plat = FALSE)
# kable(df[1:10, 1:10])
```

## Fusions

Fusions are a particular genetic event where two genes merge to create a fusion gene which is a hybrid gene formed from the two previously independent genes. It can occur as a result of translocation, interstitial deletion, or chromosomal inversion. In IMPACT datasets these can be found either in their own file or aggregated in the MAF file for mutations. In general the file containing the fusions will be very similar to a MAF file, explaining why they may be found together. We show here how to process these alterations in both cases listed above. Note that fusions are particularly rare events and thus the resulting data is very sparse.

### Processing raw fusion data

We included in `gnomeR` an example of raw downloaded MAF file directly from the website in the `fusion` dataset. We show here an example selecting the same random subset of 100 samples as in the previous section:

```{r}
as_tibble(fusion)
df <- binmat(patients = samples ,fusion = fusion)
kable(df[1:10, 1:10])
```

### Processing fusions from the API

In the API provided in `gnomeR` the fusion data comes merged with the mutation data. Note that we included an option to automatically filter them out if the user wishes so.

```{r, results="hide"}
# dat <- get_genetics(sample_ids = samples.tcga, database = 'tcga',mutations = FALSE, fusions = TRUE, cna = FALSE)
# df <- binmat(maf = dat$mut)
```

```{r}
# kable(df[1:10, 1:10])
```

Similarly to the mutation data the fusions are affected by the `specify.plat` and `set.plat` arguments as well.


## Copy-number alterations (CNA)

The final type of data we have left to cover are CNAs. This a phenomenon in which sections of the genome are repeated and the number of repeats in the genome varies between individuals. Copy number variation is a type of structural variation: specifically, it is a type of duplication or deletion event that affects a considerable number of base pairs. We will show in this section how to process CNA from IMPACT data. Once again we include an example dataset, `cna` in `gnomeR`. We will show in this section how to process raw CNA data and CNA data from the API.

The processing function for CNA is affected by two additional arguments:

- `cna.binary`: boolean declaring if the CNA data should be segregated between amplification and deletions or kept as factor variable with its original levels
- `cna.relax`: a boolean declaring if only deep deletions and full amplifications should be annotated in the case where `cna.binary` is set to FALSE. 

Note that the `specify.plat` and `set.plat` also affect CNA.

### Processing raw CNA data

By default amplifications and deletions will be separated and only deep deletions/full amplifications will accounted as shown below.

```{r}
df <- binmat(patients = samples, cna = cna)
kable(df[1:10, 1:10])
```

Setting `cna.binary` argument to FALSE yields the following events coded in a single column with their original levels:

```{r}
df <- binmat(patients = samples, cna = cna, cna.binary = FALSE)
kable(df[1:10, 1:10])
```


### Processing CNA data from the API

Setting the `cna` argument to TRUE in the `get_genetics()` function will retrieve the CNA data for the specified samples that can then further be processed in a similar fashion to the above:

```{r, results="hide"}
# dat <- get_genetics(sample_ids = samples.tcga, database = 'tcga',mutations = FALSE, fusions = FALSE, cna = TRUE)
# df <- binmat(cna = dat$cna)
```

```{r}
# kable(df[1:10, 1:10])
```

Setting `cna.binary` argument to FALSE yields the following events coded in a single column with their original levels:

```{r}
# df <- binmat(patients = samples, cna = dat$cna,cna.binary = FALSE)
# kable(df[1:10, 1:10])
```

## Processing full genetic profiles for samples

By combining all the types of data presented above, `binmat()` will provide a complete genomic profile for the specified samples. This can be done with any combination of the arguments presented above.

### Using raw data files

Once again we show how to perform this using the files directly downloaded from cBioPortal, combining the example sets `mut`, `fusion` and `cna`:

```{r}
df <- binmat(patients = samples,maf = mut, fusion = fusion, cna = cna, cna.binary = FALSE)
kable(df[1:10, c(1:3,243,244,300:305)])
```


### From the API

Similarly we get retrieve all the genetic events at once through the `get_genetics()` function and proceed to process them.

```{r, results="hide"}
# dat <- get_genetics(sample_ids = samples.tcga, database = 'tcga',mutations = TRUE, fusions = TRUE, cna = TRUE)
# df <- binmat(maf = dat$mut, cna = dat$cna,cna.binary = FALSE)
```

```{r}
# kable(df[1:10, c(1:3,243,244,300:305)])
```

Note that here the fusions are embedded in the mutation file.


# Pathway level alterations

Pathway level alteration analysis is commonly encountered in genomic studies. We therefore implemented in `binmat()` the `pathway` argument that enables the user to generate a second binary matrix with pathways as new features (columns) for each of the samples (rows). These pathways were created following Dr. Schultz's group in their paper [Oncogenic Signaling Pathways in The Cancer
Genome Atlas](https://www.cell.com/cell/pdf/S0092-8674(18)30359-3.pdf). It consist of 10 well defined pathways that have biological impacts in cancer patients.
We show below an example of this matrix. 

```{r}
df <- binmat(patients = samples,maf = mut, cna = cna,cna.binary = FALSE,pathway = T)
kable(df$pathway_dat[1:5,])
```

Note that this is returned as an additional element of `binmat()` and the original binary matrix for all genes is still returned.

## Custom pathways

In particular problems it may be of interest of creating pathways customized to the study conducted. To enable the user to use these specfic pathways we created the `custom_pathway()` function, this function takes as argument a binary output from `binmat()` with a dataframe containing the name of the genes and their corresponding pathways:

```{r}
pathway <- as.data.frame(cbind(c("PIK3CA","KRAS","TERT","TP53"),
                               c("path1","path1","path2","path3")))
pathway_dat <- custom_pathway(mat = df$mut, pathway = pathway)
kable(pathway_dat[1:5,])
```

# OncoKB annotation 

OncoKB annotates the biological and oncogenic effect and the prognostic and predictive significance of somatic molecular alterations. Potential treatment implications are stratified by the level of evidence that a specific molecular alteration is predictive of drug response based on US Food and Drug Administration (FDA) labeling, National Comprehensive Cancer Network (NCCN) guidelines, disease-focused expert group recommendations and the scientific literature. For more information see the manuscript [OncoKB: A Precision Oncology Knowledge Base](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5586540/) or the [oncoKB](https://www.oncokb.org/) website. In `gnomeR` we include a simple wrapper of the `[oncokb-annotator](https://github.com/oncokb/oncokb-annotator)` package's functions to allow users to annotate mutations, fusions and copy number events through the `oncokb()` function. Note that one of the required inputs is a token which allows users access to the oncoKB annotator. 
Users are thus required to request a token on [oncoKB](https://www.oncokb.org/). If you are an MSKCC employee simply request a token for academic use and one will be provided to you automatically.

Once the user has a token, the annotating function `oncokb()` can be used on standardized MAF, fusion and CNA files:

```{r,eval = F}
gen_oncokb <- oncokb(maf = mut, fusion = fusion, cna = cna, token = "your_token")
```

We moreover include an argument in the `binmat()` function to oncoKB annotate files while creating a binary matrix by setting oncokb to TRUE. Note that only 'Oncogenic' and 'Likely Oncogenic' variants will be kept. 

```{r,eval = F}
df <- binmat(maf = mut, fusion = fusion, cna = cna, oncokb = TRUE, token = "your_token")
```


# FACETs

The copy-number alterations data we have covered up to now is a discrete estimation of the alterations that occured. There however exist more nuanced and accurate data for copy-number alterations observed in a tumor. In `gnomeR` we include an example of segmentation file and relevant functions from the [`facets`](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/) package that provides an allele-specific copy number and clonal heterogeneity analysis tool for high-throughput DNA sequencing. We show below an example of a segmentation file included in `gnomeR` (`seg` dataset) and how to process it:

```{r}
kable(seg[1:10,])
```

We see that this files include segments of all chromosome for each patient with the number of marks and mean intensity in that segments. We can process this data into a format that can be used for visualization and analysis using the `facets.dat()` function that takes the following arguments:

- `seg`: a segmentation file
- `filenames`: the names of the segment files to be loaded and processed (Note must end in ".Rdata").
- `path`: the relative path to the files folder from your current directory
- `patients`: the names of the patients of the respective filenames. Default is using all samples available.
- `min.purity`: the minimum purity of the sample required to be kept in the final dataset. Default is 0.3.
- `epsilon`: level of unions when aggregating segments between. Default is 0.005.
- `adaptive`: CNregions option to create adaptive segments. Default is FALSE.

```{r, message = F, warning = F}
facet <- facets.dat(seg = seg, patients = samples, epsilon = 0.005)
```

This function returns a dataframe that is ready for visualization and analysis with samples as rows and processed segments as columns:

```{r, message = F, warning = F}
kable(facet$out.cn[1:5,1:3])
```
