<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data-processing • gnomeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data-processing">
<meta property="og:description" content="gnomeR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/API-tutorial.html">API-tutorial</a>
    </li>
    <li>
      <a href="../articles/Data-processing.html">Data-processing</a>
    </li>
    <li>
      <a href="../articles/gnomeR-vignette.html">gnomeR for IMPACT data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AxelitoMartin/gnomeR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data-processing</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AxelitoMartin/gnomeR/blob/master/vignettes/Data-processing.Rmd"><code>vignettes/Data-processing.Rmd</code></a></small>
      <div class="hidden name"><code>Data-processing.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gnomeR</span>)
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></pre></body></html></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The main purpose of <code>gnomeR</code> is to streamline the processing of genomic files provided by CbioPortal. If you wish to learn how to use the integrated API please read the <code>API-tutorial</code> article first. The core function of the processing of these files is performed by the <code><a href="../reference/binmat.html">binmat()</a></code> function. It takes the following arguments:</p>
<ul>
<li>patients: A character vector that let’s the user specify the patients to be used to create the matrix. Default is NULL, in which case all samples found in the provided genetic files will be used.</li>
<li>maf: A MAF file.</li>
<li>mut.type: The mutation type to be used. Options are “SOMATIC”, “GERMLINE” or “ALL”. Note “ALL” will keep all mutations regardless of status (not recommended). Default is SOMATIC.</li>
<li>SNP.only Boolean to rather the genetics events to be kept only to be SNPs (insertions and deletions will be removed). Default is FALSE.</li>
<li>include.silent: Boolean to keep or remove all silent mutations. TRUE keeps, FALSE removes. Default is FALSE.</li>
<li>fusion: An optional MAF file for fusions. If inputed the outcome will be added to the matrix with columns ending in “.fus”. Default is NULL. Note if fusions are found in the MAF file in the previous input then these will be added automatically.</li>
<li>cna: An optional CNA files. If inputed the outcome will be added to the matrix with columns ending in “.cna”, .del" and “.amp” (depending on subsequent arguments).</li>
<li>cna.binary: A boolean argument specifying if the cna events should be enforced as binary. In which case separate columns for amplifications and deletions will be created. Default is FALSE in which case columns ending in “.cna” will be added to the output.</li>
<li>cna.relax: If cna.binary is TRUE, cna data only enables to count both gains and shallow deletions as amplifications and deletions respectively.</li>
<li>spe.plat: boolean specifying if specific IMPACT platforms should be considered. When TRUE NAs will fill the cells for genes of patients that were not sequenced on that plaform. Default is TRUE.</li>
<li>set.plat character argument specifying which IMPACT platform the data should be reduced to if spe.plat is set to TRUE. Options are “341” and “410”. Default is NULL.</li>
<li>rm.empty: boolean specifying if columns with no events founds should be removed. Default is TRUE.</li>
<li>col.names character vector of the necessary columns to be used. By default: col.names = c(Tumor_Sample_Barcode = NULL, Hugo_Symbol = NULL, Variant_Classification = NULL, Mutation_Status = NULL, Variant_Type = NULL)</li>
</ul>
<p>This function returns a matrix containing all the genetic information with rows as samples and columns as features. A warning will be thrown if some samples were found to have no mutations in the MAF file.</p>
<p>In the follwing sections we will present examples to process each of the datatypes in cbioportal.</p>
</div>
<div id="processing-genetic-data" class="section level1">
<h1 class="hasAnchor">
<a href="#processing-genetic-data" class="anchor"></a>Processing genetic data</h1>
<div id="mutations" class="section level2">
<h2 class="hasAnchor">
<a href="#mutations" class="anchor"></a>Mutations</h2>
<p>The most commmon type of genetic features used in genomic studies at MSKCC. The IMPACT sequencing panel consist of a curated list of genes that are known to have cancer related properties when altered. You can find a complete list of these genes and which platform they were added on in the <code>impact_genes</code> datafile. Note that the MAF files provided through the API embedded in this package and the one provided through raw download on the website are slightly different and we will thus present examples using both these datasets.</p>
<div id="processing-raw-mutation-data" class="section level3">
<h3 class="hasAnchor">
<a href="#processing-raw-mutation-data" class="anchor"></a>Processing raw mutation data</h3>
<p>We included in <code>gnomeR</code> an example of raw downloaded MAF file directly from the website in the <code>mut</code> dataset. We show here an example selecting a random subset of 100 samples in the <code>mut</code> dataset:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu">as_tibble</span>(<span class="no">mut</span>)
<span class="co">#&gt; # A tibble: 3,179 x 45</span>
<span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id Center NCBI_Build Chromosome Start_Position</span>
<span class="co">#&gt;    &lt;fct&gt;       &lt;lgl&gt;          &lt;lgl&gt;  &lt;fct&gt;      &lt;fct&gt;               &lt;int&gt;</span>
<span class="co">#&gt;  1 TP53        NA             NA     GRCh37     17                7577539</span>
<span class="co">#&gt;  2 EZH2        NA             NA     GRCh37     7               148523605</span>
<span class="co">#&gt;  3 MDM2        NA             NA     GRCh37     12               69222656</span>
<span class="co">#&gt;  4 IGF1R       NA             NA     GRCh37     15               99486153</span>
<span class="co">#&gt;  5 KEAP1       NA             NA     GRCh37     19               10602303</span>
<span class="co">#&gt;  6 KDM5C       NA             NA     GRCh37     X                53223386</span>
<span class="co">#&gt;  7 KRAS        NA             NA     GRCh37     12               25398284</span>
<span class="co">#&gt;  8 TERT        NA             NA     GRCh37     5                 1295228</span>
<span class="co">#&gt;  9 MAP2K1      NA             NA     GRCh37     15               66729153</span>
<span class="co">#&gt; 10 NCOR1       NA             NA     GRCh37     17               16046949</span>
<span class="co">#&gt; # … with 3,169 more rows, and 39 more variables: End_Position &lt;int&gt;,</span>
<span class="co">#&gt; #   Strand &lt;fct&gt;, Consequence &lt;fct&gt;, Variant_Classification &lt;fct&gt;,</span>
<span class="co">#&gt; #   Variant_Type &lt;fct&gt;, Reference_Allele &lt;fct&gt;, Tumor_Seq_Allele1 &lt;fct&gt;,</span>
<span class="co">#&gt; #   Tumor_Seq_Allele2 &lt;fct&gt;, dbSNP_RS &lt;lgl&gt;, dbSNP_Val_Status &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Tumor_Sample_Barcode &lt;fct&gt;, Matched_Norm_Sample_Barcode &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Match_Norm_Seq_Allele1 &lt;lgl&gt;, Match_Norm_Seq_Allele2 &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Tumor_Validation_Allele1 &lt;lgl&gt;, Tumor_Validation_Allele2 &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Match_Norm_Validation_Allele1 &lt;lgl&gt;, Match_Norm_Validation_Allele2 &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Verification_Status &lt;lgl&gt;, Validation_Status &lt;lgl&gt;, Mutation_Status &lt;chr&gt;,</span>
<span class="co">#&gt; #   Sequencing_Phase &lt;lgl&gt;, Sequence_Source &lt;lgl&gt;, Validation_Method &lt;lgl&gt;,</span>
<span class="co">#&gt; #   Score &lt;lgl&gt;, BAM_File &lt;lgl&gt;, Sequencer &lt;lgl&gt;, t_ref_count &lt;int&gt;,</span>
<span class="co">#&gt; #   t_alt_count &lt;int&gt;, n_ref_count &lt;lgl&gt;, n_alt_count &lt;lgl&gt;, HGVSc &lt;fct&gt;,</span>
<span class="co">#&gt; #   HGVSp &lt;fct&gt;, HGVSp_Short &lt;fct&gt;, Transcript_ID &lt;fct&gt;, RefSeq &lt;fct&gt;,</span>
<span class="co">#&gt; #   Protein_position &lt;int&gt;, Codons &lt;fct&gt;, Hotspot &lt;int&gt;</span>
<span class="no">samples</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">mut</span>$<span class="no">Tumor_Sample_Barcode</span>))[<span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">mut</span>$<span class="no">Tumor_Sample_Barcode</span>)), <span class="fl">100</span>, <span class="kw">replace</span><span class="kw">=</span><span class="fl">FALSE</span>)]
<span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">patients</span> <span class="kw">=</span> <span class="no">samples</span> ,<span class="kw">maf</span> <span class="kw">=</span> <span class="no">mut</span>)
<span class="fu">as_tibble</span>(<span class="no">df</span>)
<span class="co">#&gt; # A tibble: 100 x 216</span>
<span class="co">#&gt;     TP53  EZH2  MDM2 IGF1R KEAP1 KDM5C  KRAS  TERT MAP2K1 NCOR1  AKT1 ARID1A</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     0     0     0     0     0     1     0      0     0     0      0</span>
<span class="co">#&gt;  2     1     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  3     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  4     0     0     0     0     0     0     1     0      0     0     0      0</span>
<span class="co">#&gt;  5     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  6     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  7     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt;  8     1     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt;  9     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt; 10     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt; # … with 90 more rows, and 204 more variables: CTNNB1 &lt;dbl&gt;, PDPK1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   CTCF &lt;dbl&gt;, CDH1 &lt;dbl&gt;, EGFR &lt;dbl&gt;, RPS6KA4 &lt;dbl&gt;, AKT3 &lt;dbl&gt;, NTRK3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ATRX &lt;dbl&gt;, STAG2 &lt;dbl&gt;, EPHA3 &lt;dbl&gt;, RECQL4 &lt;dbl&gt;, CREBBP &lt;dbl&gt;,</span>
<span class="co">#&gt; #   DOT1L &lt;dbl&gt;, PIK3CA &lt;dbl&gt;, FBXW7 &lt;dbl&gt;, APC &lt;dbl&gt;, ETV1 &lt;dbl&gt;, AXIN1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SOX9 &lt;dbl&gt;, DICER1 &lt;dbl&gt;, FGFR2 &lt;dbl&gt;, CD276 &lt;dbl&gt;, MPL &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RAD54L &lt;dbl&gt;, FLT4 &lt;dbl&gt;, ARID1B &lt;dbl&gt;, FGFR1 &lt;dbl&gt;, WT1 &lt;dbl&gt;, MLL &lt;dbl&gt;,</span>
<span class="co">#&gt; #   TBX3 &lt;dbl&gt;, FLT3 &lt;dbl&gt;, RNF43 &lt;dbl&gt;, STK11 &lt;dbl&gt;, PTPRS &lt;dbl&gt;,</span>
<span class="co">#&gt; #   DNMT3B &lt;dbl&gt;, MYD88 &lt;dbl&gt;, SETD2 &lt;dbl&gt;, SMAD3 &lt;dbl&gt;, RPTOR &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SMAD2 &lt;dbl&gt;, AXIN2 &lt;dbl&gt;, CDK6 &lt;dbl&gt;, JAK3 &lt;dbl&gt;, ESR1 &lt;dbl&gt;, PIK3CB &lt;dbl&gt;,</span>
<span class="co">#&gt; #   CCND1 &lt;dbl&gt;, SPOP &lt;dbl&gt;, RB1 &lt;dbl&gt;, ERBB2 &lt;dbl&gt;, PTPRD &lt;dbl&gt;, ATM &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERBB3 &lt;dbl&gt;, NF1 &lt;dbl&gt;, MEN1 &lt;dbl&gt;, BAP1 &lt;dbl&gt;, PBRM1 &lt;dbl&gt;, H3F3C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NOTCH2 &lt;dbl&gt;, SHQ1 &lt;dbl&gt;, LATS1 &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, RBM10 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RFWD2 &lt;dbl&gt;, HIST1H3B &lt;dbl&gt;, NOTCH4 &lt;dbl&gt;, PTCH1 &lt;dbl&gt;, IDH1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   POLE &lt;dbl&gt;, SMARCA4 &lt;dbl&gt;, INPP4A &lt;dbl&gt;, MLL2 &lt;dbl&gt;, FAT1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PLK2 &lt;dbl&gt;, BRCA2 &lt;dbl&gt;, SMAD4 &lt;dbl&gt;, BRIP1 &lt;dbl&gt;, TP63 &lt;dbl&gt;, SPEN &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RAF1 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;, CCND2 &lt;dbl&gt;, FOXP1 &lt;dbl&gt;, EP300 &lt;dbl&gt;, BCOR &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MSH2 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, GRIN2A &lt;dbl&gt;, FOXA1 &lt;dbl&gt;, GATA2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HNF1A &lt;dbl&gt;, BRAF &lt;dbl&gt;, GNAQ &lt;dbl&gt;, ASXL1 &lt;dbl&gt;, MED12 &lt;dbl&gt;, IRS1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MLH1 &lt;dbl&gt;, PDGFRA &lt;dbl&gt;, NPM1 &lt;dbl&gt;, YES1 &lt;dbl&gt;, …</span></pre></body></html></div>
<p>Note that by default in the situation above the outputted dataframe is a binary matrix made from all types of mutations and adjusting the features for the platform they were added on. Thus all samples that were sequenced on the original platform have NA’s in the cells of for features that were added on subsequent platforms. In the case where the user plans on using methods that do not accept missing values, the <code>spe.plat</code> argument can be changed to FALSE to replace all the NA’s mentioned above to 0. We show below such an example, we moreover make this example including only SNPs (including silent mutations):</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">patients</span> <span class="kw">=</span> <span class="no">samples</span> ,<span class="kw">maf</span> <span class="kw">=</span> <span class="no">mut</span>, <span class="kw">SNP.only</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">include.silent</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">spe.plat</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; Some patients did not have any mutations found in the MAF file.P-0003319-T01-</span>
<span class="co">#&gt; IM5,P-0004637-T01-IM5,P-0001585-T01-IM3,P-0006755-T01-IM5</span>
<span class="fu">as_tibble</span>(<span class="no">df</span>)
<span class="co">#&gt; # A tibble: 100 x 201</span>
<span class="co">#&gt;     TP53  EZH2  MDM2 IGF1R KEAP1 KDM5C  KRAS  TERT MAP2K1 NCOR1  AKT1 CTNNB1</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     0     0     0     0     0     1     0      0     0     0      1</span>
<span class="co">#&gt;  2     1     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt;  3     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  4     0     0     0     0     0     0     1     0      0     0     0      0</span>
<span class="co">#&gt;  5     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  6     0     0     0     0     0     0     0     1      0     0     0      0</span>
<span class="co">#&gt;  7     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt;  8     1     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt;  9     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt; 10     0     0     0     0     0     0     0     0      0     0     0      0</span>
<span class="co">#&gt; # … with 90 more rows, and 189 more variables: PDPK1 &lt;dbl&gt;, EGFR &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RPS6KA4 &lt;dbl&gt;, AKT3 &lt;dbl&gt;, NTRK3 &lt;dbl&gt;, ATRX &lt;dbl&gt;, STAG2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   EPHA3 &lt;dbl&gt;, RECQL4 &lt;dbl&gt;, CREBBP &lt;dbl&gt;, DOT1L &lt;dbl&gt;, PIK3CA &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FBXW7 &lt;dbl&gt;, APC &lt;dbl&gt;, ETV1 &lt;dbl&gt;, AXIN1 &lt;dbl&gt;, SOX9 &lt;dbl&gt;, DICER1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FGFR2 &lt;dbl&gt;, CD276 &lt;dbl&gt;, MPL &lt;dbl&gt;, RAD54L &lt;dbl&gt;, FLT4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ARID1B &lt;dbl&gt;, FGFR1 &lt;dbl&gt;, WT1 &lt;dbl&gt;, MLL &lt;dbl&gt;, TBX3 &lt;dbl&gt;, FLT3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RNF43 &lt;dbl&gt;, STK11 &lt;dbl&gt;, PTPRS &lt;dbl&gt;, DNMT3B &lt;dbl&gt;, MYD88 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SETD2 &lt;dbl&gt;, SMAD3 &lt;dbl&gt;, RPTOR &lt;dbl&gt;, SMAD2 &lt;dbl&gt;, CDK6 &lt;dbl&gt;, JAK3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ESR1 &lt;dbl&gt;, CCND1 &lt;dbl&gt;, SPOP &lt;dbl&gt;, RB1 &lt;dbl&gt;, ERBB2 &lt;dbl&gt;, PTPRD &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ATM &lt;dbl&gt;, ERBB3 &lt;dbl&gt;, NF1 &lt;dbl&gt;, BAP1 &lt;dbl&gt;, PBRM1 &lt;dbl&gt;, H3F3C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NOTCH2 &lt;dbl&gt;, SHQ1 &lt;dbl&gt;, LATS1 &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, RBM10 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RFWD2 &lt;dbl&gt;, HIST1H3B &lt;dbl&gt;, NOTCH4 &lt;dbl&gt;, PTCH1 &lt;dbl&gt;, IDH1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   POLE &lt;dbl&gt;, SMARCA4 &lt;dbl&gt;, INPP4A &lt;dbl&gt;, CDH1 &lt;dbl&gt;, MLL2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FAT1 &lt;dbl&gt;, PLK2 &lt;dbl&gt;, BRCA2 &lt;dbl&gt;, SMAD4 &lt;dbl&gt;, BRIP1 &lt;dbl&gt;, TP63 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SPEN &lt;dbl&gt;, ARID1A &lt;dbl&gt;, RAF1 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;, CCND2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FOXP1 &lt;dbl&gt;, EP300 &lt;dbl&gt;, MSH2 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, GRIN2A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GATA2 &lt;dbl&gt;, HNF1A &lt;dbl&gt;, BRAF &lt;dbl&gt;, GNAQ &lt;dbl&gt;, ASXL1 &lt;dbl&gt;, MED12 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   IRS1 &lt;dbl&gt;, MLH1 &lt;dbl&gt;, PDGFRA &lt;dbl&gt;, NPM1 &lt;dbl&gt;, YES1 &lt;dbl&gt;, ERCC2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PAK7 &lt;dbl&gt;, BCOR &lt;dbl&gt;, AR &lt;dbl&gt;, FUBP1 &lt;dbl&gt;, ROS1 &lt;dbl&gt;, …</span></pre></body></html></div>
</div>
<div id="processing-mutation-data-from-the-api" class="section level3">
<h3 class="hasAnchor">
<a href="#processing-mutation-data-from-the-api" class="anchor"></a>Processing mutation data from the API</h3>
<p>In this section we show the same examples as above but using the data downloaded through the API:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_genetics.html">get_genetics</a></span>(<span class="kw">sample_ids</span> <span class="kw">=</span> <span class="no">samples</span>, <span class="kw">fusions</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">cna</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt; [1] "https://cbioportal.mskcc.org/api/molecular-profiles/mskimpact_mutations/mutations/fetch?"</span>
<span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">maf</span> <span class="kw">=</span> <span class="no">dat</span>$<span class="no">mut</span>)
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; KMT2D has been recoded to MLL2</span>
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; KMT2C has been recoded to MLL3</span>
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; Some patients did not have any mutations found in the MAF file.P-0000788-</span>
<span class="co">#&gt; T01-IM3,P-0010898-T01-IM5,P-0008071-T01-IM5,P-0007306-T01-IM5,P-0001181-T01-</span>
<span class="co">#&gt; IM3,P-0001942-T03-IM3</span>
<span class="fu">as_tibble</span>(<span class="no">df</span>)
<span class="co">#&gt; # A tibble: 99 x 204</span>
<span class="co">#&gt;     ABL1  AKT1   ALK ALOX12B   APC BIRC3    AR  ARAF ZFHX3   ATM  ATRX   AXL</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     0     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  2     1     0     1       0     0     1     0     0     0     0     0     0</span>
<span class="co">#&gt;  3     0     1     0       0     0    NA     0     0    NA     0     0     0</span>
<span class="co">#&gt;  4     0     1     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  5     0     1     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  6     0     0     1       0     0    NA     0     0    NA     0     0     0</span>
<span class="co">#&gt;  7     0     0     0       1     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  8     0     0     0       0     1    NA     0     0    NA     0     0     0</span>
<span class="co">#&gt;  9     0     0     0       0     1    NA     0     0    NA     0     0     0</span>
<span class="co">#&gt; 10     0     0     0       0     1    NA     0     0    NA     0     0     0</span>
<span class="co">#&gt; # … with 89 more rows, and 192 more variables: B2M &lt;dbl&gt;, CCND1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   BCL6 &lt;dbl&gt;, PRDM1 &lt;dbl&gt;, BLM &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, BRAF &lt;dbl&gt;, BRCA2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   BTK &lt;dbl&gt;, CBFB &lt;dbl&gt;, CBL &lt;dbl&gt;, CCND2 &lt;dbl&gt;, CDH1 &lt;dbl&gt;, CDK4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   CDK6 &lt;dbl&gt;, CDKN2A &lt;dbl&gt;, CREBBP &lt;dbl&gt;, CSF3R &lt;dbl&gt;, CTNNB1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   DAXX &lt;dbl&gt;, DNMT3B &lt;dbl&gt;, EGFR &lt;dbl&gt;, EP300 &lt;dbl&gt;, EPHA3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   EPHA5 &lt;dbl&gt;, EPHA7 &lt;dbl&gt;, EPHB1 &lt;dbl&gt;, ERBB2 &lt;dbl&gt;, ERBB3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERBB4 &lt;dbl&gt;, ERCC2 &lt;dbl&gt;, ESR1 &lt;dbl&gt;, ETV1 &lt;dbl&gt;, EZH2 &lt;dbl&gt;, FANCA &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FANCC &lt;dbl&gt;, FAT1 &lt;dbl&gt;, FGF4 &lt;dbl&gt;, FGFR1 &lt;dbl&gt;, FGFR3 &lt;dbl&gt;, FGFR2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FGFR4 &lt;dbl&gt;, FOXO1 &lt;dbl&gt;, FLT3 &lt;dbl&gt;, FLT4 &lt;dbl&gt;, GATA2 &lt;dbl&gt;, GATA3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GNAQ &lt;dbl&gt;, GNAS &lt;dbl&gt;, GRIN2A &lt;dbl&gt;, GSK3B &lt;dbl&gt;, `HLA-A` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FOXA1 &lt;dbl&gt;, IDH1 &lt;dbl&gt;, IGF1 &lt;dbl&gt;, IGF1R &lt;dbl&gt;, INHA &lt;dbl&gt;, INHBA &lt;dbl&gt;,</span>
<span class="co">#&gt; #   INPP4A &lt;dbl&gt;, INSR &lt;dbl&gt;, IRF4 &lt;dbl&gt;, IRS1 &lt;dbl&gt;, JAK1 &lt;dbl&gt;, JAK3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   KIT &lt;dbl&gt;, KRAS &lt;dbl&gt;, SMAD2 &lt;dbl&gt;, SMAD3 &lt;dbl&gt;, SMAD4 &lt;dbl&gt;, MDM2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MAP3K1 &lt;dbl&gt;, MEN1 &lt;dbl&gt;, MET &lt;dbl&gt;, MITF &lt;dbl&gt;, MLH1 &lt;dbl&gt;, KMT2A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MPL &lt;dbl&gt;, MRE11A &lt;dbl&gt;, MSH2 &lt;dbl&gt;, NBN &lt;dbl&gt;, NF1 &lt;dbl&gt;, NF2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NFE2L2 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, NOTCH2 &lt;dbl&gt;, NOTCH3 &lt;dbl&gt;, NOTCH4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NPM1 &lt;dbl&gt;, NRAS &lt;dbl&gt;, NTRK3 &lt;dbl&gt;, PAK1 &lt;dbl&gt;, PARK2 &lt;dbl&gt;, PDGFRA &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PDGFRB &lt;dbl&gt;, PDPK1 &lt;dbl&gt;, PIK3CA &lt;dbl&gt;, PIK3CB &lt;dbl&gt;, PIK3CD &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PIK3CG &lt;dbl&gt;, PLCG2 &lt;dbl&gt;, …</span></pre></body></html></div>
<p>Similarly,</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">maf</span> <span class="kw">=</span> <span class="no">dat</span>$<span class="no">mut</span>, <span class="kw">SNP.only</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">include.silent</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">spe.plat</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; KMT2D has been recoded to MLL2</span>
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; KMT2C has been recoded to MLL3</span>
<span class="co">#&gt; Warning in createbin.maf(obj = maf, patients = patients, mut.type = mut.type, :</span>
<span class="co">#&gt; Some patients did not have any mutations found in the MAF file.P-0000788-</span>
<span class="co">#&gt; T01-IM3,P-0010898-T01-IM5,P-0008071-T01-IM5,P-0007306-T01-IM5,P-0001181-</span>
<span class="co">#&gt; T01-IM3,P-0001942-T03-IM3,P-0003319-T01-IM5,P-0006755-T01-IM5,P-0004637-T01-</span>
<span class="co">#&gt; IM5,P-0001585-T01-IM3</span>
<span class="fu">as_tibble</span>(<span class="no">df</span>)
<span class="co">#&gt; # A tibble: 99 x 192</span>
<span class="co">#&gt;     ABL1  AKT1   ALK ALOX12B   APC BIRC3    AR  ARAF ZFHX3   ATM  ATRX   AXL</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     0     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  2     0     0     1       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  3     0     1     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  4     0     1     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  5     0     0     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  6     0     0     1       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  7     0     0     0       1     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  8     0     0     0       0     1     0     0     0     0     0     0     0</span>
<span class="co">#&gt;  9     0     0     0       0     1     0     0     0     0     0     0     0</span>
<span class="co">#&gt; 10     0     0     0       0     0     0     0     0     0     0     0     0</span>
<span class="co">#&gt; # … with 89 more rows, and 180 more variables: B2M &lt;dbl&gt;, CCND1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   BCL6 &lt;dbl&gt;, PRDM1 &lt;dbl&gt;, BLM &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, BRAF &lt;dbl&gt;, BRCA2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   BTK &lt;dbl&gt;, CBL &lt;dbl&gt;, CCND2 &lt;dbl&gt;, CDH1 &lt;dbl&gt;, CDK4 &lt;dbl&gt;, CDK6 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   CDKN2A &lt;dbl&gt;, CREBBP &lt;dbl&gt;, CSF3R &lt;dbl&gt;, CTNNB1 &lt;dbl&gt;, DAXX &lt;dbl&gt;,</span>
<span class="co">#&gt; #   DNMT3B &lt;dbl&gt;, EGFR &lt;dbl&gt;, EP300 &lt;dbl&gt;, EPHA3 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   EPHA7 &lt;dbl&gt;, EPHB1 &lt;dbl&gt;, ERBB2 &lt;dbl&gt;, ERBB3 &lt;dbl&gt;, ERBB4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERCC2 &lt;dbl&gt;, ESR1 &lt;dbl&gt;, ETV1 &lt;dbl&gt;, EZH2 &lt;dbl&gt;, FANCA &lt;dbl&gt;, FANCC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FAT1 &lt;dbl&gt;, FGF4 &lt;dbl&gt;, FGFR1 &lt;dbl&gt;, FGFR3 &lt;dbl&gt;, FGFR2 &lt;dbl&gt;, FGFR4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FOXO1 &lt;dbl&gt;, FLT3 &lt;dbl&gt;, FLT4 &lt;dbl&gt;, GATA2 &lt;dbl&gt;, GATA3 &lt;dbl&gt;, GNAQ &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GNAS &lt;dbl&gt;, GRIN2A &lt;dbl&gt;, GSK3B &lt;dbl&gt;, `HLA-A` &lt;dbl&gt;, FOXA1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   IDH1 &lt;dbl&gt;, IGF1R &lt;dbl&gt;, INHA &lt;dbl&gt;, INHBA &lt;dbl&gt;, INPP4A &lt;dbl&gt;, IRF4 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   IRS1 &lt;dbl&gt;, JAK1 &lt;dbl&gt;, JAK3 &lt;dbl&gt;, KIT &lt;dbl&gt;, KRAS &lt;dbl&gt;, SMAD2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SMAD3 &lt;dbl&gt;, SMAD4 &lt;dbl&gt;, MDM2 &lt;dbl&gt;, MAP3K1 &lt;dbl&gt;, MEN1 &lt;dbl&gt;, MET &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MITF &lt;dbl&gt;, MLH1 &lt;dbl&gt;, KMT2A &lt;dbl&gt;, MPL &lt;dbl&gt;, MRE11A &lt;dbl&gt;, MSH2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NBN &lt;dbl&gt;, NF1 &lt;dbl&gt;, NFE2L2 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, NOTCH2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NOTCH3 &lt;dbl&gt;, NOTCH4 &lt;dbl&gt;, NPM1 &lt;dbl&gt;, NRAS &lt;dbl&gt;, NTRK3 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PAK1 &lt;dbl&gt;, PARK2 &lt;dbl&gt;, PDGFRA &lt;dbl&gt;, PDGFRB &lt;dbl&gt;, PDPK1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PIK3CA &lt;dbl&gt;, PIK3CD &lt;dbl&gt;, PIK3CG &lt;dbl&gt;, POLE &lt;dbl&gt;, MAP2K1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PTCH1 &lt;dbl&gt;, PTPN11 &lt;dbl&gt;, PTPRD &lt;dbl&gt;, PTPRS &lt;dbl&gt;, …</span></pre></body></html></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
