<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>gnomeR for IMPACT data • gnomeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="gnomeR for IMPACT data">
<meta property="og:description" content="gnomeR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gnomeR-vignette.html">gnomeR for IMPACT data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AxelitoMartin/gnomeR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gnomeR-vignette_files/htmlwidgets-1.5.1/htmlwidgets.js"></script><script src="gnomeR-vignette_files/plotly-binding-4.9.2/plotly.js"></script><script src="gnomeR-vignette_files/typedarray-0.1/typedarray.min.js"></script><script src="gnomeR-vignette_files/jquery-1.11.3/jquery.min.js"></script><link href="gnomeR-vignette_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="gnomeR-vignette_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><link href="gnomeR-vignette_files/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet">
<script src="gnomeR-vignette_files/plotly-main-1.52.2/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>gnomeR for IMPACT data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AxelitoMartin/gnomeR/blob/master/vignettes/gnomeR-vignette.Rmd"><code>vignettes/gnomeR-vignette.Rmd</code></a></small>
      <div class="hidden name"><code>gnomeR-vignette.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"AxelitoMartin/gnomeR"</span>)</pre></body></html></div>
</div>
<div id="requirements" class="section level1">
<h1 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h1>
<p><i>gnomeR</i> requires the following packages - <code>ComplexHeatmap</code>, <code>iClusterPlus</code>, <code>cluster</code> (installed with gnomeR).</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gnomeR</span>)</pre></body></html></div>
</div>
<div id="introduction" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><i>gnomeR</i> is a R package that aims to process and analyze genetic data from cBioPortal. We include in this package the mutation, copy number alteration (CNA), fusion and clin.patientsical information of all publicly available data from cBioPortal.</p>
<div id="mutations" class="section level2">
<h2 class="hasAnchor">
<a href="#mutations" class="anchor"></a>Mutations</h2>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="no">mut</span>) <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="no">Tumor_Sample_Barcode</span>,<span class="no">Hugo_Symbol</span>,<span class="no">Variant_Classification</span>,<span class="no">Variant_Type</span>,<span class="no">Reference_Allele</span>,<span class="no">Tumor_Seq_Allele2</span>)
<span class="co">#&gt; # A tibble: 3,179 x 6</span>
<span class="co">#&gt;    Tumor_Sample_Ba… Hugo_Symbol Variant_Classif… Variant_Type Reference_Allele</span>
<span class="co">#&gt;    &lt;fct&gt;            &lt;fct&gt;       &lt;fct&gt;            &lt;fct&gt;        &lt;fct&gt;           </span>
<span class="co">#&gt;  1 P-0000062-T01-I… TP53        Missense_Mutati… SNP          G               </span>
<span class="co">#&gt;  2 P-0000062-T01-I… EZH2        Missense_Mutati… SNP          G               </span>
<span class="co">#&gt;  3 P-0000062-T01-I… MDM2        Missense_Mutati… SNP          A               </span>
<span class="co">#&gt;  4 P-0000062-T01-I… IGF1R       Missense_Mutati… SNP          T               </span>
<span class="co">#&gt;  5 P-0000062-T01-I… KEAP1       Missense_Mutati… SNP          G               </span>
<span class="co">#&gt;  6 P-0000062-T01-I… KDM5C       Missense_Mutati… SNP          G               </span>
<span class="co">#&gt;  7 P-0000065-T01-I… KRAS        Missense_Mutati… SNP          C               </span>
<span class="co">#&gt;  8 P-0000065-T01-I… TERT        5'Flank          SNP          G               </span>
<span class="co">#&gt;  9 P-0000065-T01-I… MAP2K1      Missense_Mutati… SNP          T               </span>
<span class="co">#&gt; 10 P-0000065-T01-I… NCOR1       Nonsense_Mutati… SNP          C               </span>
<span class="co">#&gt; # … with 3,169 more rows, and 1 more variable: Tumor_Seq_Allele2 &lt;fct&gt;</span></pre></body></html></div>
</div>
<div id="cna" class="section level2">
<h2 class="hasAnchor">
<a href="#cna" class="anchor"></a>CNA</h2>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="no">cna</span>[<span class="fl">1</span>:<span class="fl">5</span>,<span class="fl">1</span>:<span class="fl">5</span>])
<span class="co">#&gt; # A tibble: 5 x 5</span>
<span class="co">#&gt;   Hugo_Symbol P.0001890.T01.I… P.0000569.T01.I… P.0000795.T01.I…</span>
<span class="co">#&gt;   &lt;fct&gt;                  &lt;int&gt;            &lt;int&gt;            &lt;int&gt;</span>
<span class="co">#&gt; 1 ABL1                       0                0                0</span>
<span class="co">#&gt; 2 ACVR1                      0                0                0</span>
<span class="co">#&gt; 3 AKT1                       0                0                0</span>
<span class="co">#&gt; 4 AKT2                       0                0                0</span>
<span class="co">#&gt; 5 AKT3                       0                0                0</span>
<span class="co">#&gt; # … with 1 more variable: P.0002060.T01.IM3 &lt;int&gt;</span></pre></body></html></div>
</div>
<div id="fusions" class="section level2">
<h2 class="hasAnchor">
<a href="#fusions" class="anchor"></a>Fusions</h2>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="no">fusion</span>) <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="no">Tumor_Sample_Barcode</span>,<span class="no">Hugo_Symbol</span>,<span class="no">Fusion</span>)
<span class="co">#&gt; # A tibble: 127 x 3</span>
<span class="co">#&gt;    Tumor_Sample_Barcode Hugo_Symbol Fusion             </span>
<span class="co">#&gt;    &lt;fct&gt;                &lt;fct&gt;       &lt;fct&gt;              </span>
<span class="co">#&gt;  1 P-0010011-T01-IM5    PAX8        PAX8-intragenic    </span>
<span class="co">#&gt;  2 P-0010977-T01-IM5    TFE3        ASPSCR1-TFE3 fusion</span>
<span class="co">#&gt;  3 P-0010977-T01-IM5    ASPSCR1     ASPSCR1-TFE3 fusion</span>
<span class="co">#&gt;  4 P-0010398-T01-IM5    BRAF        OSBPL9-BRAF fusion </span>
<span class="co">#&gt;  5 P-0010398-T01-IM5    OSBPL9      OSBPL9-BRAF fusion </span>
<span class="co">#&gt;  6 P-0010177-T01-IM5    ALK         EML4-ALK fusion    </span>
<span class="co">#&gt;  7 P-0010177-T01-IM5    EML4        EML4-ALK fusion    </span>
<span class="co">#&gt;  8 P-0010604-T01-IM5    MLL3        MLL3-intragenic    </span>
<span class="co">#&gt;  9 P-0010794-T01-IM5    ERG         TMPRSS2-ERG fusion </span>
<span class="co">#&gt; 10 P-0010794-T01-IM5    TMPRSS2     TMPRSS2-ERG fusion </span>
<span class="co">#&gt; # … with 117 more rows</span></pre></body></html></div>
</div>
<div id="clinical" class="section level2">
<h2 class="hasAnchor">
<a href="#clinical" class="anchor"></a>Clinical</h2>
<div id="patients-information" class="section level3">
<h3 class="hasAnchor">
<a href="#patients-information" class="anchor"></a>Patients information</h3>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">clin.patients</span>))
<span class="co">#&gt; # A tibble: 6 x 6</span>
<span class="co">#&gt;   X.Patient.Ident… Sex   Patient.s.Vital… Smoking.History Overall.Surviva…</span>
<span class="co">#&gt;   &lt;fct&gt;            &lt;fct&gt; &lt;fct&gt;            &lt;fct&gt;           &lt;fct&gt;           </span>
<span class="co">#&gt; 1 P-0000062        Male  DECEASED         Never           23.44           </span>
<span class="co">#&gt; 2 P-0000065        Fema… DECEASED         Prev/Curr Smok… 12.69           </span>
<span class="co">#&gt; 3 P-0000080        Fema… DECEASED         Never           &lt;NA&gt;            </span>
<span class="co">#&gt; 4 P-0000088        Male  DECEASED         Never           26.13           </span>
<span class="co">#&gt; 5 P-0000096        Male  DECEASED         Prev/Curr Smok… 17.62           </span>
<span class="co">#&gt; 6 P-0000113        Fema… ALIVE            Prev/Curr Smok… 31.85           </span>
<span class="co">#&gt; # … with 1 more variable: Overall.Survival.Status &lt;fct&gt;</span></pre></body></html></div>
</div>
<div id="samples-information" class="section level3">
<h3 class="hasAnchor">
<a href="#samples-information" class="anchor"></a>Samples information</h3>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">clin.sample</span>))
<span class="co">#&gt; # A tibble: 6 x 16</span>
<span class="co">#&gt;   X.Patient.Ident… Sample.Identifi… Sample.Collecti… Specimen.Preser…</span>
<span class="co">#&gt;   &lt;fct&gt;            &lt;fct&gt;            &lt;fct&gt;            &lt;fct&gt;           </span>
<span class="co">#&gt; 1 P-0002656        P-0002656-T01-I… Outside          FFPE            </span>
<span class="co">#&gt; 2 P-0001680        P-0001680-T02-I… Outside          FFPE            </span>
<span class="co">#&gt; 3 P-0001741        P-0001741-T01-I… Outside          FFPE            </span>
<span class="co">#&gt; 4 P-0002640        P-0002640-T01-I… Outside          FFPE            </span>
<span class="co">#&gt; 5 P-0002091        P-0002091-T01-I… In-House         FFPE            </span>
<span class="co">#&gt; 6 P-0001197        P-0001197-T01-I… Outside          FFPE            </span>
<span class="co">#&gt; # … with 12 more variables: Specimen.Type &lt;fct&gt;, DNA.Input &lt;fct&gt;,</span>
<span class="co">#&gt; #   Sample.coverage &lt;fct&gt;, Tumor.Purity &lt;fct&gt;, Matched.Status &lt;fct&gt;,</span>
<span class="co">#&gt; #   Sample.Type &lt;fct&gt;, Primary.Tumor.Site &lt;fct&gt;, Metastatic.Site &lt;fct&gt;,</span>
<span class="co">#&gt; #   Sample.Class &lt;fct&gt;, Oncotree.Code &lt;fct&gt;, Cancer.Type &lt;fct&gt;,</span>
<span class="co">#&gt; #   Cancer.Type.Detailed &lt;fct&gt;</span></pre></body></html></div>
</div>
</div>
</div>
<div id="maf-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#maf-processing" class="anchor"></a>MAF processing</h1>
<p>MAF files are the standard file format for mutation information. Each line represents a single mutation mapped to a sample, a particular gene and specific effect. All these fields are required in order to properly process the file. The IMPACT platform sequences a set of targeted oncogenic genes that are on cBioportal.</p>
<div id="creating-a-binary-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-a-binary-matrix" class="anchor"></a>Creating a binary matrix</h2>
<p>We can create a binary matrix of genetic events from the files described above. If a patient has a mutation in gene X the entry will be marked with a 1, otherwise it will be a 0. This function has the following arguments:</p>
<ul>
<li>patients : a character vector of sample IDs to be matched in the MAF file. Default is NULL in which case all samples will be used.</li>
<li>maf : the corresponding MAF file (required)</li>
<li>mut.type : the mutation type to be used, options are “SOMATIC”, “GERMLINE” or “ALL”. Default is “SOMATIC”</li>
<li>SNP.only : boolean specifying if only SNPs should be used. Default is F</li>
<li>include.silent : boolean specifying if silent mutations should be included. Default is F (recommended)</li>
<li>fusion : optional fusion file to be included. Fusions will be added as their own column with “.fus” suffix</li>
<li>cna : optional copy number alteration file to be included. CNAs will be added as their column with “.Amp” suffix for amplifications and “.Del” suffix for deletions</li>
<li>cna.relax : boolean specifying if shallow deletions and low level gains should be counted as deletions and amplifications respectively</li>
<li>spe.plat : boolean specifying if specific IMPACT platforms should be considered. When TRUE NAs will fill the cells for genes of patients that were not sequenced on that plaform</li>
<li>set.plat : character argument specifying which IMPACT platform the data should be reduced to if spe.plat is set to TRUE. Options are “341” and “410”. Default is NULL.</li>
<li>rm.empty : boolean specifying if columns with no events founds should be removed. Default is TRUE.</li>
</ul>
<p>This function will return a binary matrix of genetics events with patients as rows and columns as genes. Along with a list of patients that weren’t found to have any events (if any).</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">patients</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">mut</span>$<span class="no">Tumor_Sample_Barcode</span>))[<span class="fl">1</span>:<span class="fl">200</span>]
<span class="no">bin.mut</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">patients</span> <span class="kw">=</span> <span class="no">patients</span>,<span class="kw">maf</span> <span class="kw">=</span> <span class="no">mut</span>,<span class="kw">mut.type</span> <span class="kw">=</span> <span class="st">"SOMATIC"</span>,<span class="kw">SNP.only</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">include.silent</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">spe.plat</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu">as.tbl</span>(<span class="no">bin.mut</span>)
<span class="co">#&gt; # A tibble: 200 x 275</span>
<span class="co">#&gt;     TP53  EZH2  MDM2 IGF1R KEAP1 KDM5C  KRAS  TERT MAP2K1 NCOR1  DDR2  IDH1</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     1     1     1     1     1     0     0      0     0     0     0</span>
<span class="co">#&gt;  2     0     0     0     0     0     0     1     1      1     1     0     0</span>
<span class="co">#&gt;  3     1     0     0     0     0     0     0     0      0     0     1     1</span>
<span class="co">#&gt;  4     0     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt;  5     1     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt;  6     0     0     0     0     0     1     1     0      0     1     0     0</span>
<span class="co">#&gt;  7     1     0     0     0     0     0     1     0      0     0     0     0</span>
<span class="co">#&gt;  8     1     0     0     1     0     0     0     1      0     0     0     0</span>
<span class="co">#&gt;  9     0     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt; 10     1     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt; # … with 190 more rows, and 263 more variables: FIP1L1 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RASA1 &lt;dbl&gt;, RAD50 &lt;dbl&gt;, ESR1 &lt;dbl&gt;, HGF &lt;dbl&gt;, MLL3 &lt;dbl&gt;, RET &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FLT1 &lt;dbl&gt;, BRCA2 &lt;dbl&gt;, RB1 &lt;dbl&gt;, PTPRT &lt;dbl&gt;, AR &lt;dbl&gt;, FOXL2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FLT4 &lt;dbl&gt;, PIK3CA &lt;dbl&gt;, EP300 &lt;dbl&gt;, FBXW7 &lt;dbl&gt;, MPL &lt;dbl&gt;, RIT1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NTRK1 &lt;dbl&gt;, FAT1 &lt;dbl&gt;, RICTOR &lt;dbl&gt;, TNFAIP3 &lt;dbl&gt;, ARID1B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GNAQ &lt;dbl&gt;, SYK &lt;dbl&gt;, TSC1 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, PAK1 &lt;dbl&gt;, MLL2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   POLE &lt;dbl&gt;, TRAF7 &lt;dbl&gt;, NF1 &lt;dbl&gt;, GNA11 &lt;dbl&gt;, INSR &lt;dbl&gt;, AXL &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERG &lt;dbl&gt;, ARID1A &lt;dbl&gt;, MLH1 &lt;dbl&gt;, EGFR &lt;dbl&gt;, TBX3 &lt;dbl&gt;, MDC1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RECQL4 &lt;dbl&gt;, SPEN &lt;dbl&gt;, NTRK2 &lt;dbl&gt;, STAG2 &lt;dbl&gt;, CREBBP &lt;dbl&gt;,</span>
<span class="co">#&gt; #   AKT1 &lt;dbl&gt;, CTNNB1 &lt;dbl&gt;, PDPK1 &lt;dbl&gt;, CTCF &lt;dbl&gt;, CDH1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PPP2R1A &lt;dbl&gt;, CCND1 &lt;dbl&gt;, RUNX1 &lt;dbl&gt;, CARD11 &lt;dbl&gt;, ARID5B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERBB2 &lt;dbl&gt;, BCOR &lt;dbl&gt;, BRAF &lt;dbl&gt;, ATM &lt;dbl&gt;, MTOR &lt;dbl&gt;, STK40 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SPOP &lt;dbl&gt;, RAF1 &lt;dbl&gt;, MAP3K13 &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, CIC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RPS6KA4 &lt;dbl&gt;, AKT3 &lt;dbl&gt;, NTRK3 &lt;dbl&gt;, ATRX &lt;dbl&gt;, PBRM1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ASXL2 &lt;dbl&gt;, SF3B1 &lt;dbl&gt;, PIK3CB &lt;dbl&gt;, TET2 &lt;dbl&gt;, CDKN2B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MRE11A &lt;dbl&gt;, CBL &lt;dbl&gt;, TSHR &lt;dbl&gt;, BLM &lt;dbl&gt;, CBFB &lt;dbl&gt;, DOT1L &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RBM10 &lt;dbl&gt;, SMAD3 &lt;dbl&gt;, CDK8 &lt;dbl&gt;, GATA2 &lt;dbl&gt;, STK11 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SMARCA4 &lt;dbl&gt;, PIK3C2G &lt;dbl&gt;, HIST1H3B &lt;dbl&gt;, MYOD1 &lt;dbl&gt;, APC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   EPHA3 &lt;dbl&gt;, ETV6 &lt;dbl&gt;, PTCH1 &lt;dbl&gt;, TET1 &lt;dbl&gt;, BMPR1A &lt;dbl&gt;, …</span></pre></body></html></div>
<p>Similarly including fusions and CNAs:</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">bin.mut</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/binmat.html">binmat</a></span>(<span class="kw">patients</span> <span class="kw">=</span> <span class="no">patients</span>,<span class="kw">maf</span> <span class="kw">=</span> <span class="no">mut</span>,<span class="kw">mut.type</span> <span class="kw">=</span> <span class="st">"SOMATIC"</span>,<span class="kw">SNP.only</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">include.silent</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">fusion</span> <span class="kw">=</span> <span class="no">fusion</span>, <span class="kw">cna</span> <span class="kw">=</span> <span class="no">cna</span>, <span class="kw">spe.plat</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu">as.tbl</span>(<span class="no">bin.mut</span>)
<span class="co">#&gt; # A tibble: 200 x 522</span>
<span class="co">#&gt;     TP53  EZH2  MDM2 IGF1R KEAP1 KDM5C  KRAS  TERT MAP2K1 NCOR1  DDR2  IDH1</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1     1     1     1     1     1     1     0     0      0     0     0     0</span>
<span class="co">#&gt;  2     0     0     0     0     0     0     1     1      1     1     0     0</span>
<span class="co">#&gt;  3     1     0     0     0     0     0     0     0      0     0     1     1</span>
<span class="co">#&gt;  4     0     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt;  5     1     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt;  6     0     0     0     0     0     1     1     0      0     1     0     0</span>
<span class="co">#&gt;  7     1     0     0     0     0     0     1     0      0     0     0     0</span>
<span class="co">#&gt;  8     1     0     0     1     0     0     0     1      0     0     0     0</span>
<span class="co">#&gt;  9     0     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt; 10     1     0     0     0     0     0     0     0      0     0     0     0</span>
<span class="co">#&gt; # … with 190 more rows, and 510 more variables: FIP1L1 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RASA1 &lt;dbl&gt;, RAD50 &lt;dbl&gt;, ESR1 &lt;dbl&gt;, HGF &lt;dbl&gt;, MLL3 &lt;dbl&gt;, RET &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FLT1 &lt;dbl&gt;, BRCA2 &lt;dbl&gt;, RB1 &lt;dbl&gt;, PTPRT &lt;dbl&gt;, AR &lt;dbl&gt;, FOXL2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   FLT4 &lt;dbl&gt;, PIK3CA &lt;dbl&gt;, EP300 &lt;dbl&gt;, FBXW7 &lt;dbl&gt;, MPL &lt;dbl&gt;, RIT1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   NTRK1 &lt;dbl&gt;, FAT1 &lt;dbl&gt;, RICTOR &lt;dbl&gt;, TNFAIP3 &lt;dbl&gt;, ARID1B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GNAQ &lt;dbl&gt;, SYK &lt;dbl&gt;, TSC1 &lt;dbl&gt;, NOTCH1 &lt;dbl&gt;, PAK1 &lt;dbl&gt;, MLL2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   POLE &lt;dbl&gt;, TRAF7 &lt;dbl&gt;, NF1 &lt;dbl&gt;, GNA11 &lt;dbl&gt;, INSR &lt;dbl&gt;, AXL &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERG &lt;dbl&gt;, ARID1A &lt;dbl&gt;, MLH1 &lt;dbl&gt;, EGFR &lt;dbl&gt;, TBX3 &lt;dbl&gt;, MDC1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RECQL4 &lt;dbl&gt;, SPEN &lt;dbl&gt;, NTRK2 &lt;dbl&gt;, STAG2 &lt;dbl&gt;, CREBBP &lt;dbl&gt;,</span>
<span class="co">#&gt; #   AKT1 &lt;dbl&gt;, CTNNB1 &lt;dbl&gt;, PDPK1 &lt;dbl&gt;, CTCF &lt;dbl&gt;, CDH1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   PPP2R1A &lt;dbl&gt;, CCND1 &lt;dbl&gt;, RUNX1 &lt;dbl&gt;, CARD11 &lt;dbl&gt;, ARID5B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ERBB2 &lt;dbl&gt;, BCOR &lt;dbl&gt;, BRAF &lt;dbl&gt;, ATM &lt;dbl&gt;, MTOR &lt;dbl&gt;, STK40 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SPOP &lt;dbl&gt;, RAF1 &lt;dbl&gt;, MAP3K13 &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, CIC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RPS6KA4 &lt;dbl&gt;, AKT3 &lt;dbl&gt;, NTRK3 &lt;dbl&gt;, ATRX &lt;dbl&gt;, PBRM1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ASXL2 &lt;dbl&gt;, SF3B1 &lt;dbl&gt;, PIK3CB &lt;dbl&gt;, TET2 &lt;dbl&gt;, CDKN2B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MRE11A &lt;dbl&gt;, CBL &lt;dbl&gt;, TSHR &lt;dbl&gt;, BLM &lt;dbl&gt;, CBFB &lt;dbl&gt;, DOT1L &lt;dbl&gt;,</span>
<span class="co">#&gt; #   RBM10 &lt;dbl&gt;, SMAD3 &lt;dbl&gt;, CDK8 &lt;dbl&gt;, GATA2 &lt;dbl&gt;, STK11 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   SMARCA4 &lt;dbl&gt;, PIK3C2G &lt;dbl&gt;, HIST1H3B &lt;dbl&gt;, MYOD1 &lt;dbl&gt;, APC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   EPHA3 &lt;dbl&gt;, ETV6 &lt;dbl&gt;, PTCH1 &lt;dbl&gt;, TET1 &lt;dbl&gt;, BMPR1A &lt;dbl&gt;, …</span></pre></body></html></div>
</div>
<div id="visualizing-genetics" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-genetics" class="anchor"></a>Visualizing genetics</h2>
<p>We include a function to visualize summaries of the mutations in a given cohort:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="../reference/maf.summary.html">maf.summary</a></span>(<span class="kw">maf</span> <span class="kw">=</span> <span class="no">mut</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">Tumor_Sample_Barcode</span> <span class="kw">%in%</span> <span class="no">patients</span>),
            <span class="kw">mut.type</span> <span class="kw">=</span> <span class="st">"SOMATIC"</span>)
<span class="co">#&gt; $p.class</span></pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.type</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-2.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.SNV</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-3.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.patient.variant</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-4.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.variant.bp</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-5.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.genes</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-6.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.variant.dist</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-7.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.variant.dist.bar</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-8.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.SNV.dist</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-9.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.corr</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-10.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $p.comut</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-11-11.png" width="768"></p>
</div>
</div>
<div id="correlating-genetics-with-outcome" class="section level1">
<h1 class="hasAnchor">
<a href="#correlating-genetics-with-outcome" class="anchor"></a>Correlating genetics with outcome</h1>
<div id="binary-outcome" class="section level2">
<h2 class="hasAnchor">
<a href="#binary-outcome" class="anchor"></a>Binary outcome</h2>
<p>The <code>gen.tab</code> function allows us to test for potential differences in genetic event frequencies using Fisher’s exact test for unpaired data and the McNemar exact test for paired data. This function takes the following arguments:</p>
<ul>
<li>gen.dat : a binary matrix of genetic events with patients as rows and columns as genes</li>
<li>outcome : a binary vector corresponding to the patients in the gen.dat argument</li>
<li>filter : a value in [0,1) that will remove all genes that have a frequency below that threshold. Default is 0, including all</li>
<li>paired : boolean specifying is the observations are paired. Default is false</li>
<li>cont : boolean specifying if the outcome is continuous. Default is false (see following section for an example)</li>
<li>rank : boolean specifying if the output table should be ranked by pvalue</li>
</ul>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">outcome</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">clin.sample</span>$<span class="no">Sample.Type</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">patients</span>,<span class="no">clin.sample</span>$<span class="no">Sample.Identifier</span>)])
<span class="no">gen.dat</span> <span class="kw">&lt;-</span> <span class="no">bin.mut</span>
<span class="fu"><a href="../reference/gen.tab.html">gen.tab</a></span>(<span class="kw">gen.dat</span> <span class="kw">=</span> <span class="no">gen.dat</span>,
        <span class="kw">outcome</span> <span class="kw">=</span> <span class="no">outcome</span>,
        <span class="kw">filter</span> <span class="kw">=</span> <span class="fl">0.05</span>,<span class="kw">paired</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">cont</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">rank</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="co">#&gt; $fits</span>
<span class="co">#&gt;            Overall Metastasis(N=94) Primary(N=106) OddsRatio   Pvalue Lower</span>
<span class="co">#&gt; MYC.Amp         5%            9.57%          0.94%      0.09 6.84e-03     0</span>
<span class="co">#&gt; CREBBP          5%            8.51%          1.89%      0.21 4.82e-02  0.02</span>
<span class="co">#&gt; CDKN2A.Del    5.5%            2.13%          8.49%      4.24 6.34e-02  0.85</span>
<span class="co">#&gt; FAT1            8%            11.7%          4.72%      0.38 1.15e-01   0.1</span>
<span class="co">#&gt; CCND1.Amp     5.5%            8.51%          2.83%      0.31 1.19e-01  0.05</span>
<span class="co">#&gt; EGFR          8.5%            11.7%          5.66%      0.45 1.37e-01  0.13</span>
<span class="co">#&gt; BRAF            5%            7.45%          2.83%      0.36 1.95e-01  0.06</span>
<span class="co">#&gt; ARID1A          7%            9.57%          4.72%      0.47 2.67e-01  0.12</span>
<span class="co">#&gt; ATRX          7.5%            5.32%          9.43%      1.85 2.97e-01  0.55</span>
<span class="co">#&gt; PTEN            5%            3.19%           6.6%      2.14 3.40e-01  0.47</span>
<span class="co">#&gt; CDKN2A        5.5%            7.45%          3.77%      0.49 3.54e-01   0.1</span>
<span class="co">#&gt; SMARCA4         5%            6.38%          3.77%      0.58 5.21e-01  0.12</span>
<span class="co">#&gt; MLL2            7%            8.51%          5.66%      0.65 5.80e-01  0.18</span>
<span class="co">#&gt; TP53           47%           48.94%         45.28%      0.86 6.71e-01  0.48</span>
<span class="co">#&gt; TERT         13.5%           14.89%         12.26%       0.8 6.80e-01  0.32</span>
<span class="co">#&gt; ATM             5%            4.26%          5.66%      1.35 7.52e-01  0.31</span>
<span class="co">#&gt; MLL3          6.5%            7.45%          5.66%      0.75 7.75e-01   0.2</span>
<span class="co">#&gt; MCL1.Amp      6.5%            7.45%          5.66%      0.75 7.75e-01   0.2</span>
<span class="co">#&gt; APC           8.5%            7.45%          9.43%      1.29 8.00e-01  0.42</span>
<span class="co">#&gt; KRAS         13.5%           12.77%         14.15%      1.13 8.38e-01  0.46</span>
<span class="co">#&gt; KEAP1           6%            6.38%          5.66%      0.88 1.00e+00  0.23</span>
<span class="co">#&gt; RB1             7%            7.45%           6.6%      0.88 1.00e+00  0.25</span>
<span class="co">#&gt; PIK3CA       11.5%            11.7%         11.32%      0.96 1.00e+00  0.37</span>
<span class="co">#&gt; NOTCH1        6.5%            6.38%           6.6%      1.04 1.00e+00  0.29</span>
<span class="co">#&gt; NF1             7%            7.45%           6.6%      0.88 1.00e+00  0.25</span>
<span class="co">#&gt; STK11           6%            6.38%          5.66%      0.88 1.00e+00  0.23</span>
<span class="co">#&gt;            Upper      FDR</span>
<span class="co">#&gt; MYC.Amp     0.68 1.78e-01</span>
<span class="co">#&gt; CREBBP      1.08 5.49e-01</span>
<span class="co">#&gt; CDKN2A.Del 41.39 5.49e-01</span>
<span class="co">#&gt; FAT1        1.23 5.94e-01</span>
<span class="co">#&gt; CCND1.Amp   1.36 5.94e-01</span>
<span class="co">#&gt; EGFR        1.41 5.94e-01</span>
<span class="co">#&gt; BRAF        1.65 7.24e-01</span>
<span class="co">#&gt; ARID1A      1.63 8.37e-01</span>
<span class="co">#&gt; ATRX        7.17 8.37e-01</span>
<span class="co">#&gt; PTEN       13.19 8.37e-01</span>
<span class="co">#&gt; CDKN2A         2 8.37e-01</span>
<span class="co">#&gt; SMARCA4     2.52 1.00e+00</span>
<span class="co">#&gt; MLL2        2.22 1.00e+00</span>
<span class="co">#&gt; TP53        1.56 1.00e+00</span>
<span class="co">#&gt; TERT        1.96 1.00e+00</span>
<span class="co">#&gt; ATM         6.71 1.00e+00</span>
<span class="co">#&gt; MLL3        2.71 1.00e+00</span>
<span class="co">#&gt; MCL1.Amp    2.71 1.00e+00</span>
<span class="co">#&gt; APC         4.19 1.00e+00</span>
<span class="co">#&gt; KRAS         2.8 1.00e+00</span>
<span class="co">#&gt; KEAP1       3.42 1.00e+00</span>
<span class="co">#&gt; RB1         3.07 1.00e+00</span>
<span class="co">#&gt; PIK3CA      2.55 1.00e+00</span>
<span class="co">#&gt; NOTCH1      3.89 1.00e+00</span>
<span class="co">#&gt; NF1         3.07 1.00e+00</span>
<span class="co">#&gt; STK11       3.42 1.00e+00</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $forest.plot</span></pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $vPlot</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-12-2.png" width="768"></p>
</div>
<div id="continuous-outcome" class="section level2">
<h2 class="hasAnchor">
<a href="#continuous-outcome" class="anchor"></a>Continuous outcome</h2>
<p>Similarly we show here an example with a simulated continuous outcome:</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">1</span>)
<span class="no">outcome</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(<span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">gen.dat</span>))
<span class="no">tab.out</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/gen.tab.html">gen.tab</a></span>(<span class="kw">gen.dat</span> <span class="kw">=</span> <span class="no">gen.dat</span>,
        <span class="kw">outcome</span> <span class="kw">=</span> <span class="no">outcome</span>,
        <span class="kw">filter</span> <span class="kw">=</span> <span class="fl">0.05</span>,<span class="kw">paired</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">cont</span> <span class="kw">=</span> <span class="no">T</span>,<span class="kw">rank</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">tab.out</span>$<span class="no">fits</span>
<span class="co">#&gt;                Estimate        SD     Pvalue MutationFreq       FDR</span>
<span class="co">#&gt; CREBBP      0.706453087 0.2979996 0.01871704        0.050 0.4866429</span>
<span class="co">#&gt; RB1        -0.504577135 0.2556339 0.04979212        0.070 0.6190481</span>
<span class="co">#&gt; BRAF        0.543238493 0.2997231 0.07142863        0.050 0.6190481</span>
<span class="co">#&gt; STK11      -0.392769821 0.2759241 0.15617366        0.060 0.8294738</span>
<span class="co">#&gt; KEAP1      -0.379055669 0.2760210 0.17121660        0.060 0.8294738</span>
<span class="co">#&gt; TERT        0.251565632 0.1919061 0.19141703        0.135 0.8294738</span>
<span class="co">#&gt; MCL1.Amp   -0.301757992 0.2663017 0.25852438        0.065 0.9602334</span>
<span class="co">#&gt; CCND1.Amp  -0.271367028 0.2882525 0.34763618        0.055 0.9711260</span>
<span class="co">#&gt; FAT1        0.207573598 0.2423249 0.39270581        0.080 0.9711260</span>
<span class="co">#&gt; SMARCA4    -0.239201361 0.3017207 0.42884871        0.050 0.9711260</span>
<span class="co">#&gt; APC         0.155662676 0.2359083 0.51012067        0.085 0.9711260</span>
<span class="co">#&gt; NF1         0.160854523 0.2578834 0.53351031        0.070 0.9711260</span>
<span class="co">#&gt; EGFR       -0.144320470 0.2359447 0.54145726        0.085 0.9711260</span>
<span class="co">#&gt; ATM        -0.177916874 0.3019346 0.55636103        0.050 0.9711260</span>
<span class="co">#&gt; MLL3       -0.152630613 0.2669435 0.56812554        0.065 0.9711260</span>
<span class="co">#&gt; ATRX       -0.111901482 0.2499303 0.65483698        0.075 0.9711260</span>
<span class="co">#&gt; PTEN       -0.129519584 0.3020590 0.66854290        0.050 0.9711260</span>
<span class="co">#&gt; CDKN2A      0.122321044 0.2887661 0.67231802        0.055 0.9711260</span>
<span class="co">#&gt; KRAS       -0.067382326 0.1926775 0.72692390        0.135 0.9818790</span>
<span class="co">#&gt; PIK3CA     -0.053412691 0.2064173 0.79608869        0.115 0.9818790</span>
<span class="co">#&gt; MYC.Amp     0.069205605 0.3021592 0.81907663        0.050 0.9818790</span>
<span class="co">#&gt; CDKN2A.Del -0.044260632 0.2888798 0.87838499        0.055 0.9818790</span>
<span class="co">#&gt; TP53       -0.015388047 0.1319588 0.90728532        0.470 0.9818790</span>
<span class="co">#&gt; ARID1A     -0.016111501 0.2581341 0.95029512        0.070 0.9818790</span>
<span class="co">#&gt; NOTCH1      0.013568427 0.2671621 0.95954625        0.065 0.9818790</span>
<span class="co">#&gt; MLL2       -0.005870519 0.2581363 0.98187903        0.070 0.9818790</span>
<span class="no">tab.out</span>$<span class="no">vPlot</span></pre></body></html></div>
<div id="htmlwidget-ac6052d99643cff0dab8" style="width:768px;height:384px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-ac6052d99643cff0dab8">{"x":{"visdat":{"983068583fae":["function () ","plotlyVisDat"]},"cur_data":"983068583fae","attrs":{"983068583fae":{"x":{},"y":{},"text":{},"mode":"markers","color":{},"size":{},"alpha_stroke":1,"sizes":[10,100],"spans":[1,20]}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"title":"Volcano Plot","xaxis":{"domain":[0,1],"automargin":true,"title":"Estimate"},"yaxis":{"domain":[0,1],"automargin":true,"title":"-log10(Pvalue)"},"hovermode":"closest","showlegend":false,"legend":{"yanchor":"top","y":0.5}},"source":"A","config":{"showSendToCloud":false},"data":[{"x":[-0.0153880467741214,-0.379055668968149,-0.0673823257351641,0.251565632107747,-0.15263061287434,-0.50457713454332,-0.0534126905560237,0.207573597577214,0.0135684272587205,-0.00587051946324435,0.160854523234948,-0.0161115008789996,-0.144320469821988,0.70645308671508,0.543238492686134,-0.177916873955994,-0.111901481824652,-0.392769821251894,-0.239201361026593,0.155662676437213,-0.129519583576588,0.122321044462183,-0.271367027913781,-0.044260632135584,-0.301757991706273,0.0692056045237993],"y":[0.0422561176565427,0.766454138614024,0.138511052412042,0.71801942166836,0.24555568401527,1.30283939249894,0.0990385465271865,0.405932669877314,0.0179340880729368,0.00794201721997372,0.272857181496432,0.0221415007428096,0.266435816093429,1.72776294332861,1.14612766381849,0.254643297769975,0.18386680482417,0.806392216137775,0.367695896497227,0.292327079654482,0.174870718604305,0.172425247297193,0.458875035836477,0.0563150961421839,0.587498500188867,0.0866754644265514],"text":["Gene : TP53 <\/br> Estimate : -0.02","Gene : KEAP1 <\/br> Estimate : -0.38","Gene : KRAS <\/br> Estimate : -0.07","Gene : TERT <\/br> Estimate : 0.25","Gene : MLL3 <\/br> Estimate : -0.15","Gene : RB1 <\/br> Estimate : -0.5","Gene : PIK3CA <\/br> Estimate : -0.05","Gene : FAT1 <\/br> Estimate : 0.21","Gene : NOTCH1 <\/br> Estimate : 0.01","Gene : MLL2 <\/br> Estimate : -0.01","Gene : NF1 <\/br> Estimate : 0.16","Gene : ARID1A <\/br> Estimate : -0.02","Gene : EGFR <\/br> Estimate : -0.14","Gene : CREBBP <\/br> Estimate : 0.71","Gene : BRAF <\/br> Estimate : 0.54","Gene : ATM <\/br> Estimate : -0.18","Gene : ATRX <\/br> Estimate : -0.11","Gene : STK11 <\/br> Estimate : -0.39","Gene : SMARCA4 <\/br> Estimate : -0.24","Gene : APC <\/br> Estimate : 0.16","Gene : PTEN <\/br> Estimate : -0.13","Gene : CDKN2A <\/br> Estimate : 0.12","Gene : CCND1.Amp <\/br> Estimate : -0.27","Gene : CDKN2A.Del <\/br> Estimate : -0.04","Gene : MCL1.Amp <\/br> Estimate : -0.3","Gene : MYC.Amp <\/br> Estimate : 0.07"],"mode":"markers","type":"scatter","marker":{"colorbar":{"title":"Estimate","ticklen":2},"cmin":-0.50457713454332,"cmax":0.70645308671508,"colorscale":[["0","rgba(68,1,84,1)"],["0.0416666666666667","rgba(70,19,97,1)"],["0.0833333333333333","rgba(72,32,111,1)"],["0.125","rgba(71,45,122,1)"],["0.166666666666667","rgba(68,58,128,1)"],["0.208333333333333","rgba(64,70,135,1)"],["0.25","rgba(60,82,138,1)"],["0.291666666666667","rgba(56,93,140,1)"],["0.333333333333333","rgba(49,104,142,1)"],["0.375","rgba(46,114,142,1)"],["0.416666666666667","rgba(42,123,142,1)"],["0.458333333333333","rgba(38,133,141,1)"],["0.5","rgba(37,144,140,1)"],["0.541666666666667","rgba(33,154,138,1)"],["0.583333333333333","rgba(39,164,133,1)"],["0.625","rgba(47,174,127,1)"],["0.666666666666667","rgba(53,183,121,1)"],["0.708333333333333","rgba(79,191,110,1)"],["0.75","rgba(98,199,98,1)"],["0.791666666666667","rgba(119,207,85,1)"],["0.833333333333333","rgba(147,214,70,1)"],["0.875","rgba(172,220,52,1)"],["0.916666666666667","rgba(199,225,42,1)"],["0.958333333333333","rgba(226,228,40,1)"],["1","rgba(253,231,37,1)"]],"showscale":false,"color":[-0.0153880467741214,-0.379055668968149,-0.0673823257351641,0.251565632107747,-0.15263061287434,-0.50457713454332,-0.0534126905560237,0.207573597577214,0.0135684272587205,-0.00587051946324435,0.160854523234948,-0.0161115008789996,-0.144320469821988,0.70645308671508,0.543238492686134,-0.177916873955994,-0.111901481824652,-0.392769821251894,-0.239201361026593,0.155662676437213,-0.129519583576588,0.122321044462183,-0.271367027913781,-0.044260632135584,-0.301757991706273,0.0692056045237993],"size":[100,12.1428571428571,28.2142857142857,28.2142857142857,13.2142857142857,14.2857142857143,23.9285714285714,16.4285714285714,13.2142857142857,14.2857142857143,14.2857142857143,14.2857142857143,17.5,10,10,10,15.3571428571429,12.1428571428571,10,17.5,10,11.0714285714286,11.0714285714286,11.0714285714286,13.2142857142857,10],"sizemode":"area","line":{"colorbar":{"title":"","ticklen":2},"cmin":-0.50457713454332,"cmax":0.70645308671508,"colorscale":[["0","rgba(68,1,84,1)"],["0.0416666666666667","rgba(70,19,97,1)"],["0.0833333333333333","rgba(72,32,111,1)"],["0.125","rgba(71,45,122,1)"],["0.166666666666667","rgba(68,58,128,1)"],["0.208333333333333","rgba(64,70,135,1)"],["0.25","rgba(60,82,138,1)"],["0.291666666666667","rgba(56,93,140,1)"],["0.333333333333333","rgba(49,104,142,1)"],["0.375","rgba(46,114,142,1)"],["0.416666666666667","rgba(42,123,142,1)"],["0.458333333333333","rgba(38,133,141,1)"],["0.5","rgba(37,144,140,1)"],["0.541666666666667","rgba(33,154,138,1)"],["0.583333333333333","rgba(39,164,133,1)"],["0.625","rgba(47,174,127,1)"],["0.666666666666667","rgba(53,183,121,1)"],["0.708333333333333","rgba(79,191,110,1)"],["0.75","rgba(98,199,98,1)"],["0.791666666666667","rgba(119,207,85,1)"],["0.833333333333333","rgba(147,214,70,1)"],["0.875","rgba(172,220,52,1)"],["0.916666666666667","rgba(199,225,42,1)"],["0.958333333333333","rgba(226,228,40,1)"],["1","rgba(253,231,37,1)"]],"showscale":false,"color":[-0.0153880467741214,-0.379055668968149,-0.0673823257351641,0.251565632107747,-0.15263061287434,-0.50457713454332,-0.0534126905560237,0.207573597577214,0.0135684272587205,-0.00587051946324435,0.160854523234948,-0.0161115008789996,-0.144320469821988,0.70645308671508,0.543238492686134,-0.177916873955994,-0.111901481824652,-0.392769821251894,-0.239201361026593,0.155662676437213,-0.129519583576588,0.122321044462183,-0.271367027913781,-0.044260632135584,-0.301757991706273,0.0692056045237993]}},"textfont":{"size":[100,12.1428571428571,28.2142857142857,28.2142857142857,13.2142857142857,14.2857142857143,23.9285714285714,16.4285714285714,13.2142857142857,14.2857142857143,14.2857142857143,14.2857142857143,17.5,10,10,10,15.3571428571429,12.1428571428571,10,17.5,10,11.0714285714286,11.0714285714286,11.0714285714286,13.2142857142857,10]},"error_y":{"width":[]},"error_x":{"width":[]},"xaxis":"x","yaxis":"y","frame":null},{"x":[-0.50457713454332,0.70645308671508],"y":[0.00794201721997372,1.72776294332861],"type":"scatter","mode":"markers","opacity":0,"hoverinfo":"none","showlegend":false,"marker":{"colorbar":{"title":"Estimate","ticklen":2,"len":0.5,"lenmode":"fraction","y":1,"yanchor":"top"},"cmin":-0.50457713454332,"cmax":0.70645308671508,"colorscale":[["0","rgba(68,1,84,1)"],["0.0416666666666667","rgba(70,19,97,1)"],["0.0833333333333333","rgba(72,32,111,1)"],["0.125","rgba(71,45,122,1)"],["0.166666666666667","rgba(68,58,128,1)"],["0.208333333333333","rgba(64,70,135,1)"],["0.25","rgba(60,82,138,1)"],["0.291666666666667","rgba(56,93,140,1)"],["0.333333333333333","rgba(49,104,142,1)"],["0.375","rgba(46,114,142,1)"],["0.416666666666667","rgba(42,123,142,1)"],["0.458333333333333","rgba(38,133,141,1)"],["0.5","rgba(37,144,140,1)"],["0.541666666666667","rgba(33,154,138,1)"],["0.583333333333333","rgba(39,164,133,1)"],["0.625","rgba(47,174,127,1)"],["0.666666666666667","rgba(53,183,121,1)"],["0.708333333333333","rgba(79,191,110,1)"],["0.75","rgba(98,199,98,1)"],["0.791666666666667","rgba(119,207,85,1)"],["0.833333333333333","rgba(147,214,70,1)"],["0.875","rgba(172,220,52,1)"],["0.916666666666667","rgba(199,225,42,1)"],["0.958333333333333","rgba(226,228,40,1)"],["1","rgba(253,231,37,1)"]],"showscale":true,"color":[-0.50457713454332,0.70645308671508],"line":{"color":"rgba(255,127,14,1)"}},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="time-to-event-outcome" class="section level2">
<h2 class="hasAnchor">
<a href="#time-to-event-outcome" class="anchor"></a>Time to event outcome</h2>
<p>We further include <code>uni.cox</code> for univariate survival analysis if time to event data is available. This function takes as inputs:</p>
<ul>
<li>X : a data frame containing the covariates to be used</li>
<li>surv.dat : a data frame containing the time and status information</li>
<li>surv.formula : survival formula of the form <code>Surv(time,status)~.</code> . Note that delayed entry is allowed of the form <code>Surv(time1,time2,status)~.</code>
</li>
<li>filter : a value in [0,1) that will remove all genes that have a frequency below that threshold. Default is 0, including all</li>
</ul>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">surv.dat</span> <span class="kw">&lt;-</span> <span class="no">clin.patients</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">X.Patient.Identifier</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/abbreviate.html">abbreviate</a></span>(<span class="no">patients</span>,<span class="kw">strict</span> <span class="kw">=</span> <span class="no">T</span>, <span class="kw">minlength</span> <span class="kw">=</span> <span class="fl">9</span>)) <span class="kw">%&gt;%</span>
  <span class="fu">select</span>(<span class="no">X.Patient.Identifier</span>,<span class="no">Overall.Survival..Months.</span>, <span class="no">Overall.Survival.Status</span>) <span class="kw">%&gt;%</span>
  <span class="fu">rename</span>(<span class="kw">DMPID</span> <span class="kw">=</span> <span class="no">X.Patient.Identifier</span>, <span class="kw">time</span> <span class="kw">=</span> <span class="no">Overall.Survival..Months.</span>,<span class="kw">status</span> <span class="kw">=</span> <span class="no">Overall.Survival.Status</span>) <span class="kw">%&gt;%</span>
  <span class="fu">mutate</span>(<span class="kw">time</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">time</span>)),
    <span class="kw">status</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="no">status</span> <span class="kw">==</span> <span class="st">"LIVING"</span>,<span class="fl">0</span>,<span class="fl">1</span>)) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">time</span>))
<span class="no">X</span> <span class="kw">&lt;-</span> <span class="no">bin.mut</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">surv.dat</span>$<span class="no">DMPID</span>,<span class="fu"><a href="https://rdrr.io/r/base/abbreviate.html">abbreviate</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">bin.mut</span>),<span class="kw">strict</span> <span class="kw">=</span> <span class="no">T</span>, <span class="kw">minlength</span> <span class="kw">=</span> <span class="fl">9</span>)),]
<span class="fu"><a href="../reference/uni.cox.html">uni.cox</a></span>(<span class="kw">X</span> <span class="kw">=</span> <span class="no">X</span>, <span class="kw">surv.dat</span> <span class="kw">=</span> <span class="no">surv.dat</span>,<span class="kw">surv.formula</span> <span class="kw">=</span> <span class="fu">Surv</span>(<span class="no">time</span>,<span class="no">status</span>)~<span class="no">.</span>,<span class="kw">filter</span> <span class="kw">=</span> <span class="fl">0.05</span>)
<span class="co">#&gt; $tab</span>
<span class="co">#&gt;             Coefficient      Pvalue MutationFrequency    Feature       FDR</span>
<span class="co">#&gt; EGFR         0.84016994 0.008213418        0.10000000       EGFR 0.2710428</span>
<span class="co">#&gt; MLL          0.94371748 0.019632556        0.06428571        MLL 0.2876587</span>
<span class="co">#&gt; RB1          0.72107855 0.034831100        0.09285714        RB1 0.2876587</span>
<span class="co">#&gt; CREBBP       0.90205238 0.034867716        0.05714286     CREBBP 0.2876587</span>
<span class="co">#&gt; MLL2         0.56298861 0.085882224        0.10000000       MLL2 0.3788785</span>
<span class="co">#&gt; TP53         0.40028485 0.087817442        0.44285714       TP53 0.3788785</span>
<span class="co">#&gt; MYC.Amp      0.78522421 0.091124178        0.05714286    MYC.Amp 0.3788785</span>
<span class="co">#&gt; ATRX        -1.20905337 0.091849333        0.07142857       ATRX 0.3788785</span>
<span class="co">#&gt; NF1          0.62398882 0.147897134        0.06428571        NF1 0.5199551</span>
<span class="co">#&gt; KRAS         0.48280269 0.157562157        0.12857143       KRAS 0.5199551</span>
<span class="co">#&gt; CDH1        -0.97210425 0.175320775        0.05714286       CDH1 0.5259623</span>
<span class="co">#&gt; CDKN2A.Del   0.55320798 0.194971232        0.05714286 CDKN2A.Del 0.5361709</span>
<span class="co">#&gt; APC         -0.65024059 0.270369373        0.07142857        APC 0.6748753</span>
<span class="co">#&gt; KEAP1        0.42011277 0.291446637        0.07857143      KEAP1 0.6748753</span>
<span class="co">#&gt; PTEN        -0.60074354 0.308691943        0.05714286       PTEN 0.6748753</span>
<span class="co">#&gt; CCND1.Amp    0.40609721 0.341421917        0.06428571  CCND1.Amp 0.6748753</span>
<span class="co">#&gt; TERT         0.29749633 0.347663039        0.13571429       TERT 0.6748753</span>
<span class="co">#&gt; MCL1.Amp    -0.28401087 0.581163974        0.05714286   MCL1.Amp 0.9392426</span>
<span class="co">#&gt; EPHA5       -0.26751294 0.603085439        0.06428571      EPHA5 0.9392426</span>
<span class="co">#&gt; PIK3CA       0.15982029 0.670554504        0.10714286     PIK3CA 0.9392426</span>
<span class="co">#&gt; NOTCH1      -0.21260472 0.679426264        0.07142857     NOTCH1 0.9392426</span>
<span class="co">#&gt; FGF19.Amp    0.21036542 0.683085510        0.05000000  FGF19.Amp 0.9392426</span>
<span class="co">#&gt; FGF3.Amp     0.21036542 0.683085510        0.05000000   FGF3.Amp 0.9392426</span>
<span class="co">#&gt; FGF4.Amp     0.21036542 0.683085510        0.05000000   FGF4.Amp 0.9392426</span>
<span class="co">#&gt; SMARCA4     -0.18352777 0.721267649        0.06428571    SMARCA4 0.9520733</span>
<span class="co">#&gt; ARID1A       0.11594954 0.785894284        0.07857143     ARID1A 0.9957052</span>
<span class="co">#&gt; POLE        -0.11830097 0.841188641        0.05000000       POLE 0.9957052</span>
<span class="co">#&gt; FAT1        -0.06671197 0.866799501        0.10714286       FAT1 0.9957052</span>
<span class="co">#&gt; MLL3        -0.06205582 0.893527316        0.06428571       MLL3 0.9957052</span>
<span class="co">#&gt; STK11        0.02001728 0.962548003        0.08571429      STK11 0.9957052</span>
<span class="co">#&gt; PTPRT       -0.01433085 0.977783322        0.05000000      PTPRT 0.9957052</span>
<span class="co">#&gt; ATM          0.01293604 0.977809472        0.07142857        ATM 0.9957052</span>
<span class="co">#&gt; BRCA2      -18.16371626 0.995705178        0.05000000      BRCA2 0.9957052</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $p</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $KM</span>
<span class="co">#&gt; $KM[[1]]</span></pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-1.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[2]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-2.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[3]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-3.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[4]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-4.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[5]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-5.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[6]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-6.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[7]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-7.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[8]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-8.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[9]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-9.png" width="768"></p>
<pre><code>#&gt; 
#&gt; $KM[[10]]</code></pre>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-14-10.png" width="768"></p>
</div>
</div>
<div id="advanced-genetic-visuals" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-genetic-visuals" class="anchor"></a>Advanced genetic visuals</h1>
<div id="oncoprints" class="section level2">
<h2 class="hasAnchor">
<a href="#oncoprints" class="anchor"></a>OncoPrints</h2>
<p>OncoPrints are a convenient way to study comutation patterns in our cohort through the <code>plot_oncoPrint</code> function. It takes as argument:</p>
<ul>
<li>gen.dat : data frame of binary genetic events as created by the <code>binmat</code> function</li>
<li>clin.patients.dat : optional data frame of clin.patientsical covariates</li>
<li>ordered : order in which patients should be printed in the OncoPrint</li>
</ul>
<p>We show here an example with the most common genes.</p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="no">gen.dat</span> <span class="kw">&lt;-</span> <span class="no">bin.mut</span>[,<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">bin.mut</span>,<span class="fl">2</span>, <span class="no">sum</span>),<span class="kw">decreasing</span> <span class="kw">=</span> <span class="no">T</span>))[<span class="fl">1</span>:<span class="fl">15</span>]]
<span class="fu"><a href="../reference/plot_oncoPrint.html">plot_oncoPrint</a></span>(<span class="no">gen.dat</span>)</pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-15-1.png" width="768"></p>
<p>Similarly we include here an example adding patients’ clinical variables:</p>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="no">clin.patients.dat</span> <span class="kw">&lt;-</span> <span class="no">clin.patients</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/abbreviate.html">abbreviate</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gen.dat</span>),<span class="kw">strict</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">minlength</span> <span class="kw">=</span> <span class="fl">9</span>),<span class="no">clin.patients</span>$<span class="no">X.Patient.Identifier</span>),] <span class="kw">%&gt;%</span>
  <span class="fu">rename</span>(<span class="kw">DMPID</span> <span class="kw">=</span> <span class="no">X.Patient.Identifier</span>, <span class="kw">Smoker</span> <span class="kw">=</span> <span class="no">Smoking.History</span>) <span class="kw">%&gt;%</span>
  <span class="fu">select</span>(<span class="no">DMPID</span>, <span class="no">Sex</span>,<span class="no">Smoker</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">DMPID</span>)) <span class="kw">%&gt;%</span>
  <span class="fu">distinct</span>(<span class="no">DMPID</span>,<span class="kw">.keep_all</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">gen.dat</span> <span class="kw">&lt;-</span> <span class="no">gen.dat</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">clin.patients.dat</span>$<span class="no">DMPID</span>,<span class="fu"><a href="https://rdrr.io/r/base/abbreviate.html">abbreviate</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gen.dat</span>),<span class="kw">strict</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">minlength</span> <span class="kw">=</span> <span class="fl">9</span>)),]
<span class="no">clin.patients.dat</span> <span class="kw">&lt;-</span> <span class="no">clin.patients.dat</span> <span class="kw">%&gt;%</span>
  <span class="kw pkg">tibble</span><span class="kw ns">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span>(<span class="st">'DMPID'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gen.dat</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">clin.patients.dat</span>)
<span class="fu"><a href="../reference/plot_oncoPrint.html">plot_oncoPrint</a></span>(<span class="kw">gen.dat</span> <span class="kw">=</span> <span class="no">gen.dat</span>,<span class="kw">clin.dat</span> <span class="kw">=</span> <span class="no">clin.patients.dat</span>)</pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-16-1.png" width="768"></p>
</div>
<div id="facets" class="section level2">
<h2 class="hasAnchor">
<a href="#facets" class="anchor"></a>FACETs</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/">FACETs</a> is a fully integrated stand-alone pipeline that includes sequencing BAM file post-processing, joint segmentation of total- and allele-specific read counts, and integer copy number calls corrected for tumor purity, ploidy and clonal heterogeneity, with comprehensive output and integrated visualization. We integrate the output of this tool to our package to visualise the copy number alteration events in our cohort. The segmentation file is now integrated to the cBioPortal and we include it in our package. The FACETs output can be visualised using the <code>facets.heatmap</code> function which takes as input:</p>
<ul>
<li>seg : a segmentation file containing all the samples of interest</li>
<li>filenames : character vector of file names in case you are given individual segmentation files for each patients</li>
<li>path : the path to the files of the filenames character vector</li>
<li>patients : DMPIDs of the samples to be included in the heatmap. Default is NULL where all patients will be used</li>
<li>min.purity : if filenames argument is provided you may select a minimum purity to be included in the heatmap. Default is 0.3</li>
<li>epsilon : the maximum Euclidean distance between adjacent probes tolerated for denying a nonredundant region. epsilon=0 is equivalent to taking the union of all unique break points across the n samples. See CNregions function iClusterPlus for more information. Default is 0.005</li>
<li>ordered : order in which patients should be printed in the heatmap. Default is NULL, where we use hierarchical clustering.</li>
<li>outcome : potential outcome of interest to be plotted along the X-axis</li>
</ul>
<p>This function returns the a heatmap and the merged segmentation dataset used to created:</p>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="no">patients.seg</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">clin.sample</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">Sample.Identifier</span> <span class="kw">%in%</span> <span class="no">patients</span>, <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">Tumor.Purity</span>)) <span class="kw">&gt;</span> <span class="fl">30</span>) <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="no">Sample.Identifier</span>)))
<span class="no">facet</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/facets.heatmap.html">facets.heatmap</a></span>(<span class="kw">seg</span> <span class="kw">=</span> <span class="no">seg</span>, <span class="kw">patients</span><span class="kw">=</span><span class="no">patients.seg</span>[<span class="fl">0</span>:<span class="fl">100</span>])
<span class="no">facet</span>$<span class="no">p</span></pre></body></html></div>
<p><img src="gnomeR-vignette_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="fu">as.tbl</span>(<span class="no">facet</span>$<span class="no">out.cn</span>)
<span class="co">#&gt; # A tibble: 100 x 610</span>
<span class="co">#&gt;    `chr1.2488138-1… `chr1.11167550-… `chr1.14192672-… `chr1.15296019-…</span>
<span class="co">#&gt;               &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;</span>
<span class="co">#&gt;  1           0.0326           0.0326           0.0326           0.0326</span>
<span class="co">#&gt;  2          -0.0198          -0.0198          -0.0198          -0.0198</span>
<span class="co">#&gt;  3           0.0939           0.0939           0.0939           0.0939</span>
<span class="co">#&gt;  4          -0.155            0.0092           0.0092           0.0092</span>
<span class="co">#&gt;  5          -0.029           -0.029           -0.029           -0.029 </span>
<span class="co">#&gt;  6           0.0431           0.0431           0.0431           0.0431</span>
<span class="co">#&gt;  7           0.113            0.113            0.113            0.113 </span>
<span class="co">#&gt;  8          -0.408           -0.0501          -0.0501          -0.0501</span>
<span class="co">#&gt;  9          -0.0555          -0.0555          -0.0555          -0.0555</span>
<span class="co">#&gt; 10          -0.0391          -0.0391          -0.0391          -0.0391</span>
<span class="co">#&gt; # … with 90 more rows, and 606 more variables: `chr1.16199471-16235896` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.16235896-17345415` &lt;dbl&gt;, `chr1.17345415-22587878` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.22587878-30535114` &lt;dbl&gt;, `chr1.30535114-35013586` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.35013586-35468429` &lt;dbl&gt;, `chr1.35468429-36411737` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.36411737-45795044` &lt;dbl&gt;, `chr1.45795044-51434461` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.51434461-59564373` &lt;dbl&gt;, `chr1.59564373-65300293` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.65300293-65301840` &lt;dbl&gt;, `chr1.65301840-66042078` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.66042078-70742063` &lt;dbl&gt;, `chr1.70742063-82939537` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.82939537-96258925` &lt;dbl&gt;, `chr1.96258925-99589424` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.99589424-108079945` &lt;dbl&gt;, `chr1.108079945-113234456` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.113234456-117273534` &lt;dbl&gt;, `chr1.117273534-118145461` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.118145461-118223984` &lt;dbl&gt;, `chr1.118223984-120199189` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.120199189-120491113` &lt;dbl&gt;, `chr1.120491113-120572570` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.120572570-120611984` &lt;dbl&gt;, `chr1.120611984-147282446` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.147282446-152330945` &lt;dbl&gt;, `chr1.152330945-155870294` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.155870294-156785626` &lt;dbl&gt;, `chr1.156785626-161284206` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.161284206-162688895` &lt;dbl&gt;, `chr1.162688895-169022903` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.169022903-175914298` &lt;dbl&gt;, `chr1.175914298-176175911` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.176175911-180810233` &lt;dbl&gt;, `chr1.180810233-185520600` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.185520600-193094295` &lt;dbl&gt;, `chr1.193094295-202188641` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.202188641-204494686` &lt;dbl&gt;, `chr1.204494686-206646614` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.206646614-206942027` &lt;dbl&gt;, `chr1.206942027-209632563` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.209632563-219991341` &lt;dbl&gt;, `chr1.219991341-226064745` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.226064745-226549202` &lt;dbl&gt;, `chr1.226549202-226589998` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.226589998-234272887` &lt;dbl&gt;, `chr1.234272887-243663067` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr1.243663067-245977996` &lt;dbl&gt;, `chr2.4717089-25100738` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.25100738-28539877` &lt;dbl&gt;, `chr2.28539877-29416439` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.29416439-34654129` &lt;dbl&gt;, `chr2.34654129-38467258` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.38467258-47596683` &lt;dbl&gt;, `chr2.47596683-47600656` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.47600656-50178151` &lt;dbl&gt;, `chr2.50178151-60170875` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.60170875-61706028` &lt;dbl&gt;, `chr2.61706028-61711156` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.61711156-72126420` &lt;dbl&gt;, `chr2.72126420-96919700` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.96919700-99136565` &lt;dbl&gt;, `chr2.99136565-99149899` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.99149899-99373720` &lt;dbl&gt;, `chr2.99373720-120048729` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.120048729-125424612` &lt;dbl&gt;, `chr2.125424612-158739584` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.158739584-173914775` &lt;dbl&gt;, `chr2.173914775-178129282` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.178129282-185198425` &lt;dbl&gt;, `chr2.185198425-189541688` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.189541688-193538820` &lt;dbl&gt;, `chr2.193538820-193562208` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.193562208-198257106` &lt;dbl&gt;, `chr2.198257106-198299710` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.198299710-202123030` &lt;dbl&gt;, `chr2.202123030-203723989` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.203723989-204836598` &lt;dbl&gt;, `chr2.204836598-212248563` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.212248563-216768839` &lt;dbl&gt;, `chr2.216768839-217736131` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr2.217736131-242800953` &lt;dbl&gt;, `chr3.1449872-12955982` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.12955982-24205251` &lt;dbl&gt;, `chr3.24205251-30733002` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.30733002-33432307` &lt;dbl&gt;, `chr3.33432307-37035097` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.37035097-37038155` &lt;dbl&gt;, `chr3.37038155-38180336` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.38180336-41265566` &lt;dbl&gt;, `chr3.41265566-43177020` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.43177020-47058664` &lt;dbl&gt;, `chr3.47058664-59789176` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.59789176-69788801` &lt;dbl&gt;, `chr3.69788801-71015124` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.71015124-71019922` &lt;dbl&gt;, `chr3.71019922-71064752` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `chr3.71064752-71247137` &lt;dbl&gt;, …</span></pre></body></html></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
