<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualizations • gnomeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualizations">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Analizing-genomic-data.html">Analizing genomic data</a>
    </li>
    <li>
      <a href="../articles/Data-processing.html">Data-processing</a>
    </li>
    <li>
      <a href="../articles/Visualizations.html">Visualizations</a>
    </li>
    <li>
      <a href="../articles/gnomeR-vignette.html">gnomeR for IMPACT data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AxelitoMartin/gnomeR">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Visualizations_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Visualizations</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AxelitoMartin/gnomeR/blob/master/vignettes/Visualizations.Rmd"><code>vignettes/Visualizations.Rmd</code></a></small>
      <div class="hidden name"><code>Visualizations.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gnomeR)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this article we will explore the different automatized plotting functions included in <code>gnomeR</code>. There are two main functions in <code>gnomeR</code> to visualize genomic data:</p>
<ul>
<li>
<code><a href="../reference/maf_viz.html">maf_viz()</a></code> provides an extensive summary of the mutations found in the MAF file provided</li>
<li>
<code><a href="../reference/plot_oncoPrint.html">plot_oncoPrint()</a></code> provides an OncoPrint that let’s the user explore the comutation (or any genetic event and clinical factors) patterns in the cohort of interest</li>
</ul>
</div>
<div id="summarizing-a-maf-file" class="section level1">
<h1 class="hasAnchor">
<a href="#summarizing-a-maf-file" class="anchor"></a>Summarizing a MAF file</h1>
<p>The <code><a href="../reference/maf_viz.html">maf_viz()</a></code> function takes the following inputs:</p>
<ul>
<li>
<code>maf</code> a MAF file</li>
<li>
<code>...</code> any arguments that belong to the <code>binmat</code> method</li>
</ul>
<p><code><a href="../reference/maf_viz.html">maf_viz()</a></code> will return a list of plots:</p>
<ul>
<li>
<code>varclass</code> summarise the variant classification counts across the MAF</li>
<li>
<code>vartype</code> summarise the variant type counts across the MAF</li>
<li>
<code>snvclass</code> summarise the SNV class counts across the MAF</li>
<li>
<code>samplevar</code> summarise the variants per sample counts across the MAF</li>
<li>
<code>topgenes</code> summarise the most frequently mutated genes counts</li>
<li>
<code>genecor</code> summarise the correlation of genes that are most frequently mutated in a heatmap</li>
</ul>
<p>An example of a subset of these plots is shown below.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>plots_summary <span class="ot">&lt;-</span> <span class="fu">maf_viz</span>(<span class="at">maf =</span> mut)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>varclass</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-1.png" width="960"></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>vartype</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-2.png" width="960"></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>snvclass</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-3.png" width="960"></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>samplevar</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-4.png" width="960"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>topgenes</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-5.png" width="960"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>plots_summary<span class="sc">$</span>genecor</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-3-6.png" width="960"></p>
</div>
<div id="oncoprints" class="section level1">
<h1 class="hasAnchor">
<a href="#oncoprints" class="anchor"></a>OncoPrints</h1>
<p>OncoPrint is a way to visualize multiple genomic alteration events by heatmap provided through the <code>ComplexHeatmap</code> package. We implemented an altered version of this function in <code>gnomeR</code> through the <code><a href="../reference/plot_oncoPrint.html">plot_oncoPrint()</a></code> function that enables the user to explore the co-occurences in genomic and clinical features selected.</p>
<p>The <code><a href="../reference/plot_oncoPrint.html">plot_oncoPrint()</a></code> function takes as arguments: - <code>gen.dat</code> a genetic dataframe from <code><a href="../reference/binmat.html">binmat()</a></code> - <code>clin.dat</code> a set of clinical factors for the patients listed in gen.dat - <code>ordered</code> a numeric vector specifying which order the samples should be plotted in (optional)</p>
<div id="genetics-only" class="section level2">
<h2 class="hasAnchor">
<a href="#genetics-only" class="anchor"></a>Genetics only</h2>
<p>In the first section we show how to use the output of the <code><a href="../reference/binmat.html">binmat()</a></code> output to create an OncoPrint for a selected subset of genes. For consistency we will use the 10 genes used in the comutation plot above.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ATM"</span>,<span class="st">"BRAF"</span>,<span class="st">"MLL2"</span>,<span class="st">"APC"</span>,<span class="st">"EGFR"</span>,<span class="st">"MLL3"</span>,<span class="st">"PIK3CA"</span>,<span class="st">"KRAS"</span>,<span class="st">"TP53"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> <span class="fu">binmat</span>(<span class="at">maf =</span> mut, <span class="at">fusion =</span> fusion, <span class="at">cna =</span> cna)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> dat.gen <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(genes))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_oncoPrint</span>(<span class="at">gen.dat =</span> dat.gen)</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-4-1.png" width="960"></p>
</div>
<div id="missing-information" class="section level2">
<h2 class="hasAnchor">
<a href="#missing-information" class="anchor"></a>Missing information</h2>
<p>A critical point of <code>gnomeR</code> is its ability to correct for the different platforms that have existed at MSKCC. We reflect this is in the OncoPrints by including black entries in the missing cells as shown below.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ATM"</span>,<span class="st">"BRAF"</span>,<span class="st">"MLL2"</span>,<span class="st">"APC"</span>,<span class="st">"EGFR"</span>,<span class="st">"MLL3"</span>,<span class="st">"PIK3CA"</span>,<span class="st">"KRAS"</span>,<span class="st">"TP53"</span>,<span class="st">"EPHA7"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> <span class="fu">binmat</span>(<span class="at">maf =</span> mut, <span class="at">fusion =</span> fusion, <span class="at">cna =</span> cna)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> dat.gen <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(genes))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_oncoPrint</span>(<span class="at">gen.dat =</span> dat.gen)</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
</div>
<div id="adding-clinical-information" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-clinical-information" class="anchor"></a>Adding clinical information</h2>
<p>In this section we will show how the user can further add clinical information to this plot.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ATM"</span>,<span class="st">"BRAF"</span>,<span class="st">"MLL2"</span>,<span class="st">"APC"</span>,<span class="st">"EGFR"</span>,<span class="st">"MLL3"</span>,<span class="st">"PIK3CA"</span>,<span class="st">"KRAS"</span>,<span class="st">"TP53"</span>,<span class="st">"EPHA7"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> <span class="fu">binmat</span>(<span class="at">maf =</span> mut, <span class="at">fusion =</span> fusion, <span class="at">cna =</span> cna)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> dat.gen <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(genes))</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>clin.patients.dat <span class="ot">&lt;-</span> clin.patients[<span class="fu">match</span>(</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abbreviate</span>(<span class="fu">rownames</span>(dat.gen),<span class="at">strict =</span> <span class="cn">TRUE</span>, <span class="at">minlength =</span> <span class="dv">9</span>),</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  clin.patients<span class="sc">$</span>X.Patient.Identifier),] <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">DMPID =</span> X.Patient.Identifier, <span class="at">Smoker =</span> Smoking.History) <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(DMPID, Sex,Smoker) <span class="sc">%&gt;%</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(DMPID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(DMPID,<span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>dat.gen <span class="ot">&lt;-</span> dat.gen[<span class="fu">match</span>(clin.patients.dat<span class="sc">$</span>DMPID,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">abbreviate</span>(<span class="fu">rownames</span>(dat.gen),<span class="at">strict =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">minlength =</span> <span class="dv">9</span>)),]</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>clin.patients.dat <span class="ot">&lt;-</span> clin.patients.dat <span class="sc">%&gt;%</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">DMPID =</span> <span class="fu">as.character</span>(DMPID)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">'DMPID'</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dat.gen) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(clin.patients.dat)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_oncoPrint</span>(<span class="at">gen.dat =</span> dat.gen,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">clin.dat =</span> clin.patients.dat)</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-6-1.png" width="960"></p>
</div>
</div>
<div id="advanced-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-visualization" class="anchor"></a>Advanced visualization</h1>
<p>The copy-number alterations data we have covered up to now is a discrete estimation of the alterations that occured. There however exist more nuanced and accurate data for copy-number alterations observed in a tumor. In <code>gnomeR</code> we include an example of segmentation file and relevant functions from the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/"><code>facets</code></a> package that provides an allele-specific copy number and clonal heterogeneity analysis tool for high-throughput DNA sequencing. We show in this section how to visualize this data:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(mut<span class="sc">$</span>Tumor_Sample_Barcode))[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(mut<span class="sc">$</span>Tumor_Sample_Barcode)), <span class="dv">100</span>, <span class="at">replace=</span><span class="cn">FALSE</span>)]</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>p.heat <span class="ot">&lt;-</span> <span class="fu">facets.heatmap</span>(<span class="at">seg =</span> seg, <span class="at">patients =</span> samples, <span class="at">min.purity =</span> <span class="dv">0</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>p.heat<span class="sc">$</span>p</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
<p>In the heatmap above we show the estimate gains and losses over segments in each chromosomes (rows) and in each patient (columns). Now that the purity of the tumor sample is an important factor in order to get reliable data. Moreover we include argument <code>ordered</code> and <code>outcome</code> that enables the user to sort the heatmap by a specific feature of interest. We show an example below with a simulated feature of outcome:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>outcome <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(samples))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>p.heat <span class="ot">&lt;-</span> <span class="fu">facets.heatmap</span>(<span class="at">seg =</span> seg, <span class="at">patients =</span> samples, <span class="at">min.purity =</span> <span class="dv">0</span>, <span class="at">outcome =</span> outcome, <span class="at">ordered =</span> <span class="fu">order</span>(outcome))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>p.heat<span class="sc">$</span>p</span></code></pre></div>
<p><img src="Visualizations_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#summarizing-a-maf-file">Summarizing a MAF file</a></li>
      <li>
<a href="#oncoprints">OncoPrints</a>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#genetics-only">Genetics only</a></li>
      <li><a href="#missing-information">Missing information</a></li>
      <li><a href="#adding-clinical-information">Adding clinical information</a></li>
      </ul>
</li>
      <li><a href="#advanced-visualization">Advanced visualization</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
